diff -r -U2 neverball-1.5.4/share/base_image.c neverball-1.5.4/share/base_image.c
--- neverball-1.5.4/share/base_image.c	2009-09-20 03:05:36.000000000 -0700
+++ neverball-1.5.4/share/base_image.c	2009-09-20 03:05:36.000000000 -0700
@@ -95,5 +95,5 @@
         }
 
-        if (!(bytep = png_malloc(readp, h * png_sizeof(png_bytep))))
+        if (!(bytep = png_malloc(readp, h * sizeof(png_bytep))))
             longjmp(png_jmpbuf(readp), -1);
 
