diff -r -U2 comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp
--- comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-03 04:42:51.000000000 -0700
+++ comm-1.9.2/mozilla/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-03 04:42:51.000000000 -0700
@@ -205,5 +205,5 @@
   NotifyNotification* notify = notify_notification_new(mAlertTitle.get(),
                                                        mAlertText.get(),
-                                                       NULL, NULL);
+                                                       NULL);
   if (!notify)
     return NS_ERROR_OUT_OF_MEMORY;
