--- mysql-4.0.2-alpha/myisam/ft_static.c (3d3620ac4946897c)	2002-07-17 21:58:04.000000000 -0400
+++ mysql-4.0.2-alpha/myisam/ft_static.c_blah (3d3620ac2d6763db)	2002-07-17 21:58:04.000000000 -0400
@@ -18,7 +18,7 @@
 
 #include "ftdefs.h"
 
-ulong ft_min_word_len=4;
+ulong ft_min_word_len=2;
 ulong ft_max_word_len=HA_FT_MAXLEN;
 ulong ft_max_word_len_for_sort=20;
 const char *ft_boolean_syntax="+ -><()~*:\"\"&|";
