diff -r -U2 loser-0.9.12/src/client/startup.c loser-0.9.12/src/client/startup.c
--- loser-0.9.12/src/client/startup.c	1999-05-16 20:49:53.000000000 -0700
+++ loser-0.9.12/src/client/startup.c	1999-05-16 20:49:53.000000000 -0700
@@ -314,11 +314,10 @@
 	    || v2 != msg.server_info.v2
 	    || v3 != msg.server_info.v3)
-	  display_msg("\n\
-** WARNING **
-This server has an incompatible version (the right version for\n\
-this client is %d.%d.%d, the server reported version %d.%d.%d).\n\
-The game may play incorrectly, or may not play at all. Use a\n\
-compatible server or proceed at your own risk.\n\
-"
+	  display_msg("\n"
+"** WARNING **"
+"This server has an incompatible version (the right version for\n"
+"this client is %d.%d.%d, the server reported version %d.%d.%d).\n"
+"The game may play incorrectly, or may not play at all. Use a\n"
+"compatible server or proceed at your own risk.\n"
 		      "", v1, v2, v3, msg.server_info.v1,
 		      msg.server_info.v2, msg.server_info.v3);
