--- setup.sh	2004-05-30 18:13:51.738125912 -0700
+++ setup.sh	2004-05-30 18:13:51.738125912 -0700
@@ -368,7 +368,5 @@
              1 "Driver configuration ($SELECTED_DRIVER)" \
              2 "Software configuration" \
-             3 "Save configuration & run configure" \
-             4 "Save configuration & exit" \
-             5 "Exit WITHOUT doing anything" 2>$TEMP
+             4 "Save configuration & exit" 2>$TEMP
 
     if test "$?" != "0"; then
