diff -r -U2 dump-0.4b36/dump/main.c dump-0.4b36/dump/main.c
--- dump-0.4b36/dump/main.c	2004-04-21 01:55:51.000000000 -0700
+++ dump-0.4b36/dump/main.c	2004-04-21 01:55:51.000000000 -0700
@@ -67,4 +67,6 @@
 #include <sys/stat.h>
 #include <bsdcompat.h>
+typedef unsigned short  sa_family_t;
+#include <linux/netlink.h>
 #include <linux/fs.h>	/* for definition of BLKFLSBUF */
 #elif defined sunos
diff -r -U2 dump-0.4b36/dump/tape.c dump-0.4b36/dump/tape.c
--- dump-0.4b36/dump/tape.c	2004-04-21 02:15:08.000000000 -0700
+++ dump-0.4b36/dump/tape.c	2004-04-21 02:15:08.000000000 -0700
@@ -61,4 +61,6 @@
 #include <sys/types.h>
 #include <sys/time.h>
+typedef unsigned short  sa_family_t;
+#include <linux/netlink.h>
 #include <linux/fs.h>	/* for definition of BLKFLSBUF */
 #include <time.h>
