diff -r -U2 id3lib-3.8.3/Makefile.in id3lib-3.8.3/Makefile.in
--- id3lib-3.8.3/Makefile.in	2003-03-01 17:23:00.000000000 -0700
+++ id3lib-3.8.3/Makefile.in	2003-03-01 17:23:00.000000000 -0700
@@ -133,6 +133,6 @@
 @ID3_NEEDZLIB_FALSE@zlib_subdir = 
 
-SUBDIRS = . m4 $(zlib_subdir) doc include id3com src examples
-DIST_SUBDIRS = . m4 zlib doc include id3com src examples prj libprj
+SUBDIRS = . m4 $(zlib_subdir) doc include id3com src
+DIST_SUBDIRS = . m4 zlib doc include id3com src prj libprj
 
 INCLUDES = @ID3LIB_DEBUG_FLAGS@
@@ -500,7 +500,5 @@
 docs-release: docs
 	-mv doc/$(docsdistdir).* .
-	-cd examples && $(MAKE) $(AM_MAKEFLAGS) clean
 	-mkdir $(docsdistdir)
-	-cp -R examples           $(docsdistdir)
 	-cp    doc/*.*            $(docsdistdir)
 	-cp -R doc/@DOX_DIR_HTML@ $(docsdistdir)
diff -r -U2 id3lib-3.8.3/configure id3lib-3.8.3/configure
--- id3lib-3.8.3/configure	2003-03-01 17:23:00.000000000 -0700
+++ id3lib-3.8.3/configure	2003-03-01 17:23:00.000000000 -0700
@@ -10297,5 +10297,4 @@
 for ac_header in \
   string                        \
-  iomanip.h                     \
 
 do
diff -r -U2 id3lib-3.8.3/include/id3/id3lib_strings.h id3lib-3.8.3/include/id3/id3lib_strings.h
--- id3lib-3.8.3/include/id3/id3lib_strings.h	2003-03-01 17:23:00.000000000 -0700
+++ id3lib-3.8.3/include/id3/id3lib_strings.h	2003-03-01 17:23:00.000000000 -0700
@@ -1,2 +1,3 @@
+#include <string.h>
 // -*- C++ -*-
 // $Id: id3lib_strings.h,v 1.4 2003/03/02 13:56:21 t1mpy Exp $
diff -r -U2 id3lib-3.8.3/include/id3/writers.h id3lib-3.8.3/include/id3/writers.h
--- id3lib-3.8.3/include/id3/writers.h	2003-03-01 17:23:00.000000000 -0700
+++ id3lib-3.8.3/include/id3/writers.h	2003-03-01 17:23:00.000000000 -0700
@@ -1,2 +1,3 @@
+#include <string.h>
 // -*- C++ -*-
 // $Id: writers.h,v 1.11 2002/07/02 22:11:16 t1mpy Exp $
