diff -r -U2 kde-workspace-4.10.5/CMakeLists.txt kde-workspace-4.10.5/CMakeLists.txt
--- kde-workspace-4.10.5/CMakeLists.txt	2013-06-28 11:27:46.665287261 -0700
+++ kde-workspace-4.10.5/CMakeLists.txt	2013-06-28 11:27:46.665287261 -0700
@@ -20,5 +20,5 @@
 include(KDE4Defaults)
 
-find_package(Strigi REQUIRED)
+find_package(Strigi)
 find_package(ZLIB REQUIRED)
 find_package(KActivities 6.0.0 REQUIRED)
diff -r -U2 kde-workspace-4.10.5/kcontrol/kfontinst/CMakeLists.txt kde-workspace-4.10.5/kcontrol/kfontinst/CMakeLists.txt
--- kde-workspace-4.10.5/kcontrol/kfontinst/CMakeLists.txt	2013-06-28 10:10:20.030387665 -0700
+++ kde-workspace-4.10.5/kcontrol/kfontinst/CMakeLists.txt	2013-06-28 10:10:20.030387665 -0700
@@ -40,5 +40,4 @@
     add_subdirectory( dbus )
     add_subdirectory( kcmfontinst ) 
-    add_subdirectory( strigi-analyzer )
     add_subdirectory( apps )
     add_subdirectory( kio ) 
diff -r -U2 kde-workspace-4.10.5/plasma/desktop/applets/kickoff/CMakeLists.txt kde-workspace-4.10.5/plasma/desktop/applets/kickoff/CMakeLists.txt
--- kde-workspace-4.10.5/plasma/desktop/applets/kickoff/CMakeLists.txt	2013-06-28 10:10:19.491411282 -0700
+++ kde-workspace-4.10.5/plasma/desktop/applets/kickoff/CMakeLists.txt	2013-06-28 10:10:19.491411282 -0700
@@ -39,5 +39,5 @@
 endif(HAVE_STRIGIDBUS)
 
-include_directories( ${STRIGI_INCLUDE_DIR} )
+# include_directories( ${STRIGI_INCLUDE_DIR} )
 
 kde4_add_library(kickoff SHARED ${libkickoff_SRCS})
