--- portmap_5beta/portmap.c.orig	2003-03-13 15:38:27.000000000 +0200
+++ portmap_5beta/portmap.c		2003-03-13 15:38:01.000000000 +0200
@@ -98,4 +98,5 @@
 extern char *strerror();
 #include <stdlib.h>
+#include <errno.h>
 
 #ifndef LOG_PERROR
@@ -125,5 +126,4 @@
 struct pmaplist *pmaplist;
 int debugging = 0;
-extern int errno;
 
 #include "pmap_check.h"
@@ -296,5 +296,5 @@
 void
 perror(what)
-	const char *what;
+	__const char *what;
 {
 
