--- digikam-doc-0.9.5.orig/configure.in
+++ digikam-doc-0.9.5/configure.in
@@ -92,8 +92,6 @@
 AC_CONFIG_FILES([ doc/nl_digikam/Makefile ])
 AC_CONFIG_FILES([ doc/nl_showfoto/Makefile ])
 AC_CONFIG_FILES([ doc/pt_BR_digikam/Makefile ])
-AC_CONFIG_FILES([ doc/pt_digikam/Makefile ])
-AC_CONFIG_FILES([ doc/pt_showfoto/Makefile ])
 AC_CONFIG_FILES([ doc/ru_digikam/Makefile ])
 AC_CONFIG_FILES([ doc/showfoto/Makefile ])
 AC_CONFIG_FILES([ doc/sv_digikam/Makefile ])
