diff -r -U2 ntp-4.2.6/util/tickadj.c ntp-4.2.6/util/tickadj.c
--- ntp-4.2.6/util/tickadj.c	2009-12-09 00:36:37.000000000 -0700
+++ ntp-4.2.6/util/tickadj.c	2009-12-09 00:36:37.000000000 -0700
@@ -22,5 +22,5 @@
 #endif /* HAVE_UNISTD_H */
 
-#ifdef HAVE___ADJTIMEX		/* Linux */
+#ifdef linux
 
 #include <sys/timex.h>
@@ -92,5 +92,5 @@
 
 	if (!errflg) {
-		if (__adjtimex(&txc) < 0)
+		if (adjtimex(&txc) < 0)
 			perror("adjtimex");
 		else if (!quiet)
@@ -147,5 +147,5 @@
 	}
     
-	if (__adjtimex(&txc) < 0)
+	if (adjtimex(&txc) < 0)
 	{
 		perror("adjtimex");
