diff -r -U2 eel-2.26.0/eel/eel-background.c eel-2.26.0/eel/eel-background.c
--- eel-2.26.0/eel/eel-background.c	2009-03-02 09:21:55.000000000 -0700
+++ eel-2.26.0/eel/eel-background.c	2009-03-02 09:21:55.000000000 -0700
@@ -369,5 +369,5 @@
 	set_image_properties (background);
 
-	background->details->background_changes_with_size = gnome_bg_changes_with_size (background->details->bg);
+	background->details->background_changes_with_size = gnome_bg_has_multiple_sizes (background->details->bg);
 	background->details->background_pixmap = gnome_bg_create_pixmap (background->details->bg,
 									 window,
@@ -748,5 +748,5 @@
 	
 	background->details->background_changes_with_size =
-		gnome_bg_changes_with_size (background->details->bg);
+		gnome_bg_has_multiple_sizes (background->details->bg);
 	
 	if (background->details->is_desktop && !in_fade) {
@@ -996,12 +996,4 @@
 }
 
-/* determine if a background is darker or lighter than average, to help clients know what
-   colors to draw on top with */
-gboolean
-eel_background_is_dark (EelBackground *background)
-{
-	return gnome_bg_is_dark (background->details->bg);
-}
-   
 /* handle dropped colors */
 void
