diff -r -U2 cyrus-sasl-2.1.22/plugins/digestmd5.c cyrus-sasl-2.1.22/plugins/digestmd5.c
--- cyrus-sasl-2.1.22/plugins/digestmd5.c	2006-05-17 09:46:17.000000000 -0700
+++ cyrus-sasl-2.1.22/plugins/digestmd5.c	2006-05-17 09:46:17.000000000 -0700
@@ -2716,9 +2716,11 @@
 #ifdef WITH_RC4
 	128,				/* max_ssf */
-#elif WITH_DES
+#else
+#ifdef WITH_DES
 	112,
 #else 
 	1,
 #endif
+#endif
 	SASL_SEC_NOPLAINTEXT
 	| SASL_SEC_NOANONYMOUS
@@ -4035,9 +4037,11 @@
 #ifdef WITH_RC4				/* mech_name */
 	128,				/* max ssf */
-#elif WITH_DES
+#else
+#ifdef WITH_DES
 	112,
 #else
 	1,
 #endif
+#endif
 	SASL_SEC_NOPLAINTEXT
 	| SASL_SEC_NOANONYMOUS
