diff -r -U2 faac-1.28/common/mp4v2/mpeg4ip.h faac-1.28/common/mp4v2/mpeg4ip.h
--- faac-1.28/common/mp4v2/mpeg4ip.h	2009-01-26 15:42:35.000000000 -0700
+++ faac-1.28/common/mp4v2/mpeg4ip.h	2009-01-26 15:42:35.000000000 -0700
@@ -121,12 +121,4 @@
 #include <sys/param.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-char *strcasestr(const char *haystack, const char *needle);
-#ifdef __cplusplus
-}
-#endif
-
 #define OPEN_RDWR O_RDWR
 #define OPEN_CREAT O_CREAT 
