diff -r -U2 ntp-4.2.4p8/include/ntp_syscall.h ntp-4.2.4p8/include/ntp_syscall.h
--- ntp-4.2.4p8/include/ntp_syscall.h	2004-02-24 22:57:56.000000000 -0700
+++ ntp-4.2.4p8/include/ntp_syscall.h	2004-02-24 22:57:56.000000000 -0700
@@ -15,4 +15,12 @@
 #endif
 
+#if defined(ADJ_NANO) && !defined(MOD_NANO)
+#define MOD_NANO ADJ_NANO
+#endif
+
+#if defined(ADJ_TAI) && !defined(MOD_TAI)
+#define MOD_TAI ADJ_TAI
+#endif
+
 #ifndef NTP_SYSCALLS_LIBC
 #ifdef NTP_SYSCALLS_STD
