diff -r -U2 gnuchess-5.07/src/common.h gnuchess-5.07/src/common.h
--- gnuchess-5.07/src/common.h	2003-06-30 04:28:38.000000000 -0700
+++ gnuchess-5.07/src/common.h	2003-06-30 04:28:38.000000000 -0700
@@ -717,5 +717,5 @@
 /* Input thread and thread function */
 #include <pthread.h>
-extern pthread_t input_thread;
+static pthread_t input_thread;
 void *input_func(void *);
 
@@ -746,5 +746,5 @@
  * input routines. The given argument is the prompt.
  */
-void (*getline) (char *);
+void (*getline_chess) (char *);
 
 #define MAXSTR 128
diff -r -U2 gnuchess-5.07/src/input.c gnuchess-5.07/src/input.c
--- gnuchess-5.07/src/input.c	2003-07-01 10:06:43.000000000 -0700
+++ gnuchess-5.07/src/input.c	2003-07-01 10:06:43.000000000 -0700
@@ -128,5 +128,5 @@
     }
     pthread_mutex_lock(&input_mutex);
-    getline(prompt);
+    getline_chess(prompt);
     input_status = INPUT_AVAILABLE;
     pthread_cond_signal(&input_cond);
@@ -174,11 +174,11 @@
 #ifdef HAVE_LIBREADLINE
   if (isatty(STDIN_FILENO)) {
-    getline = getline_readline;
+    getline_chess = getline_readline;
     using_history();
   } else {
-    getline = getline_standard;
+    getline_chess = getline_standard;
   }
 #else
-  getline = getline_standard;
+  getline_chess = getline_standard;
 #endif
   /* Start input thread */
