diff -r -U2 yiff-2.14.2/yiff/Makefile yiff-2.14.2/yiff/Makefile
--- yiff-2.14.2/yiff/Makefile	2003-07-20 14:30:02.000000000 -0700
+++ yiff-2.14.2/yiff/Makefile	2003-07-20 14:30:02.000000000 -0700
@@ -159,5 +159,5 @@
 #
 $(BIN): $(OBJ_C) $(OBJ_CPP)
-	$(CC) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB) $(LIB_DIR)
+	$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB) $(LIB_DIR)
 
 all: $(BIN)
diff -r -U2 yiff-2.14.2/yiff/Makefile.Linux yiff-2.14.2/yiff/Makefile.Linux
--- yiff-2.14.2/yiff/Makefile.Linux	2003-07-20 14:30:02.000000000 -0700
+++ yiff-2.14.2/yiff/Makefile.Linux	2003-07-20 14:30:02.000000000 -0700
@@ -159,5 +159,5 @@
 #
 $(BIN): $(OBJ_C) $(OBJ_CPP)
-	$(CC) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB) $(LIB_DIR)
+	$(CPP) $(OBJ_C) $(OBJ_CPP) -o $(BIN) $(LIB) $(LIB_DIR)
 
 all: $(BIN)
diff -r -U2 yiff-2.14.2/yiffconfig/yccb.c yiff-2.14.2/yiffconfig/yccb.c
--- yiff-2.14.2/yiffconfig/yccb.c	2003-07-20 14:30:01.000000000 -0700
+++ yiff-2.14.2/yiffconfig/yccb.c	2003-07-20 14:30:01.000000000 -0700
@@ -1303,8 +1303,6 @@
 	    CDialogGetResponse(
 "No Y Audio Mode selected!",
-"There is no Y Audio Mode currently selected to apply\n\
-values for.",
-"You must select a Y Audio Mode from the list and then
-apply its values to the Y server.",
+"There is no Y Audio Mode currently selected to apply\nvalues for.",
+"You must select a Y Audio Mode from the list and then\napply its values to the Y server.",
 		CDIALOG_ICON_ERROR,
 		CDIALOG_RESPONSE_OK | CDIALOG_RESPONSE_HELP,
diff -r -U2 yiff-2.14.2/yiffutils/ymixer.c yiff-2.14.2/yiffutils/ymixer.c
--- yiff-2.14.2/yiffutils/ymixer.c	2003-07-20 14:30:00.000000000 -0700
+++ yiff-2.14.2/yiffutils/ymixer.c	2003-07-20 14:30:00.000000000 -0700
@@ -24,24 +24,24 @@
 static void print_help(void)
 {
-	printf("\
-Usage: ymixer [mixer] [value1] [value2] [value3] [value4] [options]\n\
-\n\
-    [mixer] specifies the mixer channel device name, running without\n\
-    any arguments prints a list of available mixer names.
-\n\
-    [value#] specify the percent values (from 0.0%%, 100.0%%) to be set.\n\
-    If no [value#] is specified then the current value is printed.\n\
-\n\
-    [options] can be any of the following:\n\
-\n\
-        --recorder <address:port>    Specify which Y server to connect to.\n\
-\n\
-    Return values:\n\
-\n\
-        0       Success.\n\
-        1       General error.\n\
-        2       Cannot connect to Y server error.\n\
-        3       Systems error.\n\
-\n"
+	printf(
+"Usage: ymixer [mixer] [value1] [value2] [value3] [value4] [options]\n"
+"\n"
+"    [mixer] specifies the mixer channel device name, running without\n"
+"    any arguments prints a list of available mixer names."
+"\n"
+"    [value#] specify the percent values (from 0.0%%, 100.0%%) to be set.\n"
+"    If no [value#] is specified then the current value is printed.\n"
+"\n"
+"    [options] can be any of the following:\n"
+"\n"
+"        --recorder <address:port>    Specify which Y server to connect to.\n"
+"\n"
+"    Return values:\n"
+"\n"
+"        0       Success.\n"
+"        1       General error.\n"
+"        2       Cannot connect to Y server error.\n"
+"        3       Systems error.\n"
+"\n"
 	);
 }
