diff -r -U2 ../bash-3.0/jobs.c ../bash-3.0/jobs.c
--- ../bash-3.0/jobs.c	2005-02-16 13:31:37.502684576 -0700
+++ ../bash-3.0/jobs.c	2005-02-16 13:31:37.502684576 -0700
@@ -2477,4 +2477,5 @@
   pid_t pid;
   int call_set_current, last_stopped_job, job, children_exited, waitpid_flags;
+  static int wcontinued_not_supported = 0;
 
   call_set_current = children_exited = 0;
@@ -2490,5 +2491,13 @@
       if (sigchld || block == 0)
 	waitpid_flags |= WNOHANG;
+    retry:
+      if (wcontinued_not_supported)
+	waitpid_flags &= ~WCONTINUED;
       pid = WAITPID (-1, &status, waitpid_flags);
+      if (pid == -1 && errno == EINVAL)
+	{
+	  wcontinued_not_supported = 1;
+	  goto retry;
+	}
 
       /* The check for WNOHANG is to make sure we decrement sigchld only
Only in ../bash-3.0: jobs.c~
diff -r -U2 ../bash-3.0/tests/jobs4.sub ../bash-3.0/tests/jobs4.sub
--- ../bash-3.0/tests/jobs4.sub	2004-05-06 06:02:41.000000000 -0700
+++ ../bash-3.0/tests/jobs4.sub	2004-05-06 06:02:41.000000000 -0700
@@ -19,4 +19,4 @@
 wait
 
-cat &
+sleep 100 &
 kill -1 %% && echo i killed it || echo could not kill it
