diff -pruN libcap-2.15.orig/libcap/include/sys/capability.h libcap-2.15/libcap/include/sys/capability.h
--- libcap-2.15.orig/libcap/include/sys/capability.h	2009-04-07 14:44:37.000000000 +0200
+++ libcap-2.15/libcap/include/sys/capability.h	2009-04-08 13:30:20.000000000 +0200
@@ -22,18 +22,7 @@ extern "C" {
 #include <sys/types.h>
 #include <stdint.h>
 
-/*
- * Make sure we can be included from userland by preventing
- * capability.h from including other kernel headers
- */
-#define _LINUX_TYPES_H
-#define _LINUX_FS_H
-#define __LINUX_COMPILER_H
 #define __user
-
-typedef unsigned int __u32;
-typedef __u32 __le32;
-
 #include <linux/capability.h>
 
 /*
