diff -r -U2 dvdauthor/src/spuunmux.c dvdauthor/src/spuunmux.c
--- dvdauthor/src/spuunmux.c	2010-05-10 00:27:31.000000000 -0700
+++ dvdauthor/src/spuunmux.c	2010-05-10 00:27:31.000000000 -0700
@@ -39,4 +39,5 @@
 #include <netinet/in.h>
 
+#include <zlib.h>
 #include <png.h>
 
@@ -611,5 +612,5 @@
         return -1;
       } /*if*/
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
       {
         png_destroy_write_struct(&png_ptr, &info_ptr);
