diff -r -U2 kdelibs-4.11.0/CMakeLists.txt kdelibs-4.11.0/CMakeLists.txt
--- kdelibs-4.11.0/CMakeLists.txt	2013-08-06 14:49:15.000000000 -0700
+++ kdelibs-4.11.0/CMakeLists.txt	2013-08-06 14:49:15.000000000 -0700
@@ -86,5 +86,5 @@
     set (STRIGI_EXTRA_TEXT "")
 else (WIN32)
-    set (STRIGI_TYPE "REQUIRED")
+    set (STRIGI_TYPE "OPTIONAL")
     set (STRIGI_EXTRA_TEXT "Required by some critical kioslaves")
 endif (WIN32)
diff -r -U2 kdelibs-4.11.0/kio/kio/kfilemetainfo.cpp kdelibs-4.11.0/kio/kio/kfilemetainfo.cpp
--- kdelibs-4.11.0/kio/kio/kfilemetainfo.cpp	2013-07-09 15:37:45.000000000 -0700
+++ kdelibs-4.11.0/kio/kio/kfilemetainfo.cpp	2013-07-09 15:37:45.000000000 -0700
@@ -422,7 +422,7 @@
 }
 
-const KFileMetaInfo& KFileMetaInfo::operator= ( KFileMetaInfo const & kfmi )
+KFileMetaInfo& KFileMetaInfo::operator= ( KFileMetaInfo const & kfmi )
 {
-    return kfmi;
+    return *this;
 }
 
