diff -r -U2 glibc-2.13/nptl/sysdeps/pthread/pt-initfini.c glibc-2.13/nptl/sysdeps/pthread/pt-initfini.c
--- glibc-2.13/nptl/sysdeps/pthread/pt-initfini.c	2011-01-17 21:34:07.000000000 -0700
+++ glibc-2.13/nptl/sysdeps/pthread/pt-initfini.c	2011-01-17 21:34:07.000000000 -0700
@@ -46,4 +46,9 @@
 asm ("\n#include \"defs.h\"");
 
+asm ("\n#if defined __i686 && defined __ASSEMBLER__");
+asm ("\n#undef __i686");
+asm ("\n#define __i686 __i686");
+asm ("\n#endif");
+
 /* The initial common code ends here. */
 asm ("\n/*@HEADER_ENDS*/");
diff -r -U2 glibc-2.13/sysdeps/unix/sysv/linux/i386/sysdep.h glibc-2.13/sysdeps/unix/sysv/linux/i386/sysdep.h
--- glibc-2.13/sysdeps/unix/sysv/linux/i386/sysdep.h	2011-01-17 21:34:07.000000000 -0700
+++ glibc-2.13/sysdeps/unix/sysv/linux/i386/sysdep.h	2011-01-17 21:34:07.000000000 -0700
@@ -30,4 +30,8 @@
 #include <tls.h>
 
+#if defined __i686 && defined __ASSEMBLER__
+#undef __i686
+#define __i686 __i686
+#endif
 
 /* For Linux we can use the system call table in the header file
