--- xmonisdn/Makefile.in.	Fri Oct 19 19:47:01 2001
+++ xmonisdn/Makefile.in	Fri Oct 19 19:47:01 2001
@@ -63,6 +63,8 @@
 config:
 	@cp Makefile.in GNUmakefile
 	@xmkmf
+	@echo "Hacking the Makefile ..  (Clifford)"
+	@perl -pe 's,^(xmonisdn:.*),$$1\n\t\$$(MAKE) xmonisdn.man,' < Makefile > x && mv x Makefile
 
 clean:
 	$(MAKE) -f Makefile clean
--- capiinfo/Makefile.in	Mon Nov  8 08:59:01 1999
+++ capiinfo/Makefile.in	Mon Nov  8 08:59:01 1999
@@ -115,7 +115,7 @@
 .SUFFIXES:
 .SUFFIXES: .S .c .o .s
 $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOMAKE) --foreign Makefile
+	cd $(top_srcdir) && $(AUTOMAKE) -i --foreign Makefile
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status $(BUILT_SOURCES)
 	cd $(top_builddir) \
@@ -273,7 +273,7 @@
 	top_distdir=`cd $(distdir) && pwd`; \
 	distdir=`cd $(distdir) && pwd`; \
 	cd $(top_srcdir) \
-	  && $(AUTOMAKE) --include-deps --build-dir=$$here --srcdir-name=$(top_srcdir) --output-dir=$$top_distdir --foreign Makefile
+	  && $(AUTOMAKE) --include-deps --build-dir=$$here --srcdir-name=$(top_srcdir) --output-dir=$$top_distdir -i --foreign Makefile
 	@for file in $(DISTFILES); do \
 	  d=$(srcdir); \
 	  if test -d $$d/$$file; then \
--- xisdnload/xisdnload.c	Fri Oct 19 21:08:39 2001
+++ xisdnload/xisdnload.c	Fri Oct 19 21:08:39 2001
@@ -40,6 +40,7 @@
 #include <sys/ioctl.h>
 #include <sys/fcntl.h>
 #include <sys/time.h>
+#include <time.h>
 #include <sys/types.h>
 #include <linux/isdn.h>
 
--- capifax/Makefile.in	Fri Oct 19 21:14:22 2001
+++ capifax/Makefile.in	Fri Oct 19 21:14:22 2001
@@ -124,7 +124,7 @@
 .SUFFIXES:
 .SUFFIXES: .S .c .o .s
 $(srcdir)/Makefile.in: Makefile.am $(top_srcdir)/configure.in $(ACLOCAL_M4) 
-	cd $(top_srcdir) && $(AUTOMAKE) --foreign Makefile
+	cd $(top_srcdir) && $(AUTOMAKE) -i --foreign Makefile
 
 Makefile: $(srcdir)/Makefile.in  $(top_builddir)/config.status $(BUILT_SOURCES)
 	cd $(top_builddir) \
@@ -286,7 +286,7 @@
 	top_distdir=`cd $(distdir) && pwd`; \
 	distdir=`cd $(distdir) && pwd`; \
 	cd $(top_srcdir) \
-	  && $(AUTOMAKE) --include-deps --build-dir=$$here --srcdir-name=$(top_srcdir) --output-dir=$$top_distdir --foreign Makefile
+	  && $(AUTOMAKE) --include-deps --build-dir=$$here --srcdir-name=$(top_srcdir) --output-dir=$$top_distdir -i --foreign Makefile
 	@for file in $(DISTFILES); do \
 	  d=$(srcdir); \
 	  if test -d $$d/$$file; then \
