diff -r -U2 ati_firegl-4_3_0-8.23.7-1/lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c ati_firegl-4_3_0-8.23.7-1/lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c
--- ati_firegl-4_3_0-8.23.7-1/lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c	2006-03-07 07:52:43.000000000 -0700
+++ ati_firegl-4_3_0-8.23.7-1/lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c	2006-03-13 15:42:06.000000000 -0700
@@ -22,4 +22,5 @@
 #endif /* !MODULE */
 
+#define _LINUX_UNISTD_H_
 // ============================================================
 #include <linux/version.h>
@@ -1121,5 +1122,5 @@
 int ATI_API_CALL __ke_verify_area(int type, const void * addr, unsigned long size)
 {
-    return verify_area(type, addr, size);
+    return access_ok(type,addr,size) ? 0 : -EFAULT;
 }
 
