diff -r -U2 samba-3.0.4/source/client/smbmount.c samba-3.0.4/source/client/smbmount.c
--- samba-3.0.4/source/client/smbmount.c	2004-04-20 13:42:58.000000000 -0700
+++ samba-3.0.4/source/client/smbmount.c	2004-04-20 13:42:58.000000000 -0700
@@ -25,4 +25,6 @@
 #include <mntent.h>
 #include <asm/types.h>
+#include <linux/preempt.h>
+#include <linux/netlink.h>
 #include <linux/smb_fs.h>
 
diff -r -U2 samba-3.0.4/source/client/smbumount.c samba-3.0.4/source/client/smbumount.c
--- samba-3.0.4/source/client/smbumount.c	2004-04-04 00:37:28.000000000 -0700
+++ samba-3.0.4/source/client/smbumount.c	2004-04-04 00:37:28.000000000 -0700
@@ -14,4 +14,6 @@
 #include <linux/smb.h>
 #include <linux/smb_mount.h>
+#include <linux/preempt.h>
+#include <linux/netlink.h>
 #include <linux/smb_fs.h>
 
