diff -r -U2 xterm-235/main.c fixed/xterm-235/main.c
--- xterm-235/main.c	2008-02-27 17:28:00.000000000 -0700
+++ xterm-235/main.c	2008-02-27 17:28:00.000000000 -0700
@@ -2466,5 +2466,5 @@
     /* GNU libc 2 allows us to abstract away from having to know the
        master pty device name. */
-    if ((*pty = getpt()) >= 0) {
+    if ((*pty = posix_openpt( O_RDWR | O_NOCTTY )) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {	/* if filesystem is trashed, this may be null */
