diff -r -U2 gobject-introspection-0.6.10/gir/Makefile.am gobject-introspection-0.6.10/gir/Makefile.am
--- gobject-introspection-0.6.10/gir/Makefile.am	2010-04-16 07:58:23.000000000 -0700
+++ gobject-introspection-0.6.10/gir/Makefile.am	2010-04-16 07:58:23.000000000 -0700
@@ -8,5 +8,4 @@
 
 BASE_GIRSOURCES =               \
-        cairo-1.0.gir           \
         fontconfig-2.0.gir      \
         freetype2-2.0.gir       \
@@ -157,27 +156,6 @@
 TYPELIB_VERSION = $(LT_CURRENT).0
 
-lib_LTLIBRARIES = libgirepository-everything-1.0.la
-
-libgirepository_everything_1_0_la_SOURCES = $(srcdir)/everything.c $(srcdir)/everything.h
-libgirepository_everything_1_0_la_CFLAGS = $(GOBJECT_CFLAGS)
-libgirepository_everything_1_0_la_LDFLAGS = -version-info $(LT_VERSION)
-libgirepository_everything_1_0_la_LIBADD = $(GOBJECT_LIBS) -lcairo
-
-if OS_WIN32
-libgirepository_everything_1_0_la_LDFLAGS += -no-undefined
-endif
-
-Everything-1.0.gir: Gio-2.0.gir libgirepository-everything-1.0.la
-
-Everything_1_0_gir_LIBS = libgirepository-everything-1.0.la
-Everything_1_0_gir_PACKAGES = gobject-2.0
-Everything_1_0_gir_INCLUDES = GObject-2.0 cairo-1.0
-Everything_1_0_gir_FILES = $(srcdir)/everything.h $(srcdir)/everything.c
-
-INTROSPECTION_GIRS += Everything-$(TYPELIB_VERSION).gir
-EXPECTEDGIRS += Everything-$(TYPELIB_VERSION)-expected.gir
-
 # marshalling tests
-lib_LTLIBRARIES += libgirepository-gimarshallingtests-1.0.la
+lib_LTLIBRARIES = libgirepository-gimarshallingtests-1.0.la
 
 libgirepository_gimarshallingtests_1_0_la_SOURCES = $(srcdir)/gimarshallingtests.c $(srcdir)/gimarshallingtests.h
