diff -r -U2 gobject-introspection-0.6.10/gir/Makefile.am gobject-introspection-0.6.10/gir/Makefile.am
--- gobject-introspection-0.6.10/gir/Makefile.am	2010-05-06 22:51:15.077287478 -0700
+++ gobject-introspection-0.6.10/gir/Makefile.am	2010-05-06 22:51:15.077287478 -0700
@@ -10,9 +10,5 @@
         fontconfig-2.0.gir      \
         freetype2-2.0.gir       \
-        GL-1.0.gir              \
-        libxml2-2.0.gir         \
-        xft-2.0.gir             \
-        xlib-2.0.gir            \
-        xfixes-4.0.gir
+        libxml2-2.0.gir
 EXTRA_DIST += $(BASE_GIRSOURCES)
 
