diff -r -U2 guile-1.8.4/libguile/guile.c guile-1.8.4/libguile/guile.c
--- guile-1.8.4/libguile/guile.c	2006-02-12 06:42:51.000000000 -0700
+++ guile-1.8.4/libguile/guile.c	2006-02-12 06:42:51.000000000 -0700
@@ -69,6 +69,4 @@
 #if !defined (__MINGW32__)
   /* libtool automagically inserts this variable into your executable... */
-  extern const lt_dlsymlist lt_preloaded_symbols[];
-  lt_dlpreload_default (lt_preloaded_symbols);
 #endif
   scm_boot_guile (argc, argv, inner_main, 0);
