diff -r -U2 prboom-2.5.0/config.h prboom-2.5.0/config.h
--- prboom-2.5.0/config.h	2008-11-09 12:22:46.000000000 -0700
+++ prboom-2.5.0/config.h	2008-11-09 12:22:46.000000000 -0700
@@ -10,5 +10,5 @@
 
 /* Define to be the path where Doom WADs are stored */
-#define DOOMWADDIR "/usr/local/share/games/doom"
+#define DOOMWADDIR "/usr/share/games/doom"
 
 /* Define if you are building with OpenGL support */
diff -r -U2 prboom-2.5.0/src/Makefile.am prboom-2.5.0/src/Makefile.am
--- prboom-2.5.0/src/Makefile.am	2008-10-21 23:37:23.000000000 -0700
+++ prboom-2.5.0/src/Makefile.am	2008-10-21 23:37:23.000000000 -0700
@@ -9,5 +9,5 @@
 SUBDIRS = SDL POSIX MAC
 
-gamesdir=$(prefix)/games
+gamesdir=$(prefix)/bin
 games_PROGRAMS = prboom prboom-game-server
 
Only in prboom-2.5.0/src: Makefile.am.orig
diff -r -U2 prboom-2.5.0/src/Makefile.in prboom-2.5.0/src/Makefile.in
--- prboom-2.5.0/src/Makefile.in	2008-11-09 12:22:16.000000000 -0700
+++ prboom-2.5.0/src/Makefile.in	2008-11-09 12:22:16.000000000 -0700
@@ -247,5 +247,5 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = SDL POSIX MAC
-gamesdir = $(prefix)/games
+gamesdir = $(prefix)/bin
 prboom_game_server_SOURCES = d_server.c protocol.h
 prboom_game_server_LDADD = POSIX/libposixdoom.a SDL/i_network.o @NET_LIBS@ @SDL_LIBS@
Only in prboom-2.5.0/src: Makefile.in.orig
