diff -r -U2 -N uqm-0.6.0/sc2/build.vars uqm-0.6.0/sc2/build.vars
--- uqm-0.6.0/sc2/build.vars	2006-12-15 23:21:26.000000000 -0700
+++ uqm-0.6.0/sc2/build.vars	2006-12-15 23:21:26.000000000 -0700
@@ -30,9 +30,9 @@
 REZ=''
 WINDRES=''
-uqm_CFLAGS=' -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -O3 -DGFXMODULE_SDL -DHAVE_OPENGL  -DSOUNDMODULE_SDL -pthread -D_REENTRANT   -DNETPLAY=NETPLAY_FULL -DHAVE_JOYSTICK -DHAVE_ZIP=1  -DUSE_PLATFORM_ACCEL -I "."'
-uqm_LDFLAGS=' -L/usr/lib64 -lSDL -lpthread -L/usr/lib64 -lSDL -lpthread -lSDL_image -lGL -L/usr/lib64 -lmikmod -ldl -lvorbisfile -lvorbis -lm -logg   -lz'
-uqm_INSTALL_BINDIR='/usr/local/games/bin/'
-uqm_INSTALL_LIBDIR='/usr/local/games/lib/'
-uqm_INSTALL_SHAREDIR='/usr/local/games/share/'
+uqm_CFLAGS='-Os -pipe -march=pentium3 -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -O3 -DGFXMODULE_SDL -DHAVE_OPENGL  -DSOUNDMODULE_SDL -pthread -D_REENTRANT   -DNETPLAY=NETPLAY_FULL -DHAVE_JOYSTICK -DHAVE_ZIP=1  -I "."'
+uqm_LDFLAGS='-Wl,-s -L/usr/lib -Wl,-rpath,/usr/lib -lSDL -lpthread -L/usr/lib -Wl,-rpath,/usr/lib -lSDL -lpthread -lSDL_image -lGL -L/usr/lib -lmikmod -ldl -lvorbisfile -lvorbis -lm -logg   -lz'
+uqm_INSTALL_BINDIR='/usr/bin/'
+uqm_INSTALL_LIBDIR='/usr/lib/'
+uqm_INSTALL_SHAREDIR='/usr/share/'
 
 # Exported variables are available from all the Makeinfo files
@@ -43,5 +43,5 @@
 uqm_HAVE_OPENGL='1'
 uqm_USE_ZIP_IO='1'
-uqm_USE_PLATFORM_ACCEL='1'
+uqm_USE_PLATFORM_ACCEL='0'
 uqm_NETPLAY='FULL'
 uqm_USE_WINSOCK=''
diff -r -U2 -N uqm-0.6.0/sc2/config.state uqm-0.6.0/sc2/config.state
--- uqm-0.6.0/sc2/config.state	2006-12-15 23:21:25.000000000 -0700
+++ uqm-0.6.0/sc2/config.state	2006-12-15 23:21:25.000000000 -0700
@@ -7,6 +7,6 @@
 CHOICE_joystick_VALUE='enabled'
 CHOICE_ioformat_VALUE='stdio_zip'
-CHOICE_accel_VALUE='asm'
-INPUT_install_prefix_VALUE='/usr/local/games'
+CHOICE_accel_VALUE='plainc'
+INPUT_install_prefix_VALUE='/usr'
 INPUT_install_bindir_VALUE='$prefix/bin'
 INPUT_install_libdir_VALUE='$prefix/lib'
diff -r -U2 -N uqm-0.6.0/sc2/config_unix.h uqm-0.6.0/sc2/config_unix.h
--- uqm-0.6.0/sc2/config_unix.h	2006-12-15 23:21:25.000000000 -0700
+++ uqm-0.6.0/sc2/config_unix.h	2006-12-15 23:21:25.000000000 -0700
@@ -12,5 +12,5 @@
 
 /* Directory where the UQM game data is located */
-#define CONTENTDIR "/usr/local/games/share/uqm/content"
+#define CONTENTDIR "/usr/share/uqm/content"
 
 /* Directory where game data will be stored */
diff -r -U2 -N uqm-0.6.0/sc2/uqm-wrapper uqm-0.6.0/sc2/uqm-wrapper
--- uqm-0.6.0/sc2/uqm-wrapper	2006-12-15 23:21:26.000000000 -0700
+++ uqm-0.6.0/sc2/uqm-wrapper	2006-12-15 23:21:26.000000000 -0700
@@ -1,4 +1,4 @@
 #!/bin/sh
 # Wrapper script for starting The Ur-Quan Masters
-exec "/usr/local/games/lib/uqm/uqm" "--contentdir=/usr/local/games/share/uqm/content" "$@"
+exec "/usr/lib/uqm/uqm" "--contentdir=/usr/share/uqm/content" "$@"
 
