diff -r -U2 multipath-tools-0.4.9/multipath.conf.annotated multipath-tools-0.4.9/multipath.conf.annotated
--- multipath-tools-0.4.9/multipath.conf.annotated	2010-05-22 05:01:58.000000000 -0700
+++ multipath-tools-0.4.9/multipath.conf.annotated	2010-05-22 05:01:58.000000000 -0700
@@ -241,14 +241,15 @@
 ## default : fd, hd, md, dm, sr, scd, st, ram, raw, loop, dcssblk
 ##
-#blacklist {
-#       wwid 26353900f02796769
-#	devnode "^(ram|raw|loop|fd|md|dm-|sr|scd|st)[0-9]*"
-#	devnode "^hd[a-z]"
-#	devnode "^dcssblk[0-9]*"
-#       device {
-#               vendor DEC.*
-#               product MSA[15]00
-#       }
-#}
+blacklist {
+       wwid 26353900f02796769
+	devnode "^(ram|raw|loop|fd|md|dm-|sr|scd|st)[0-9]*"
+	devnode "^hd[a-z]"
+	devnode "^sd[a-z]"
+	devnode "^dcssblk[0-9]*"
+       device {
+               vendor DEC.*
+               product MSA[15]00
+       }
+}
 ##
 ## name    : blacklist_exceptions
