--- php-5.2.0/ext/curl/interface.c	2006-10-10 16:12:59.000000000 -0700
+++ php-5.2.0/ext/curl/interface.c	2006-10-10 16:12:59.000000000 -0700
@@ -3,5 +3,5 @@
    | PHP Version 5                                                        |
    +----------------------------------------------------------------------+
-   | Copyright (c) 1997-2006 The PHP Group                                |
+   | Copyright (c) 1997-2007 The PHP Group                                |
    +----------------------------------------------------------------------+
    | This source file is subject to version 3.01 of the PHP license,      |
@@ -17,5 +17,5 @@
 */
 
-/* $Id: interface.c,v 1.62.2.14.2.12 2006/10/10 23:12:59 iliaa Exp $ */
+/* $Id: interface.c,v 1.62.2.14.2.22 2007/01/19 18:03:33 tony2001 Exp $ */
 
 #define ZEND_INCLUDE_FULL_WINDOWS_HEADERS
@@ -157,5 +157,11 @@
 #define CAAZ(s, v) add_assoc_zval_ex(return_value, s, sizeof(s), (zval *) v);
 
-#define PHP_CURL_CHECK_OPEN_BASEDIR(str, len)													\
+#if defined(PHP_WIN32) || defined(__GNUC__)
+ #define php_curl_ret(__ret) RETVAL_FALSE; return __ret;
+#else
+ #define php_curl_ret(__ret) RETVAL_FALSE; return;
+#endif
+
+#define PHP_CURL_CHECK_OPEN_BASEDIR(str, len, __ret)													\
 	if (((PG(open_basedir) && *PG(open_basedir)) || PG(safe_mode)) &&                                                \
 	    strncasecmp(str, "file:", sizeof("file:") - 1) == 0)								\
@@ -165,10 +171,11 @@
 		if (!(tmp_url = php_url_parse_ex(str, len))) {											\
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Invalid URL '%s'", str);				\
-			RETURN_FALSE; 																		\
-		} 																						\
+			php_curl_ret(__ret);											\
+		} 													\
 															\
 		if (php_memnstr(str, tmp_url->path, strlen(tmp_url->path), str + len)) {				\
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "URL '%s' contains unencoded control characters.", str);	\
-			RETURN_FALSE;											\
+			php_url_free(tmp_url); 																\
+			php_curl_ret(__ret);											\
 		}													\
 																								\
@@ -177,5 +184,5 @@
 		) { 																					\
 			php_url_free(tmp_url); 																\
-			RETURN_FALSE; 																		\
+			php_curl_ret(__ret);											\
 		} 																						\
 		php_url_free(tmp_url); 																	\
@@ -370,5 +377,7 @@
 	REGISTER_CURL_CONSTANT(CURLOPT_NETRC);
 	REGISTER_CURL_CONSTANT(CURLOPT_FOLLOWLOCATION);
+#if CURLOPT_FTPASCII != 0
 	REGISTER_CURL_CONSTANT(CURLOPT_FTPASCII);
+#endif
 	REGISTER_CURL_CONSTANT(CURLOPT_PUT);
 #if CURLOPT_MUTE != 0
@@ -410,5 +419,7 @@
 	REGISTER_CURL_CONSTANT(CURLOPT_WRITEFUNCTION);
 	REGISTER_CURL_CONSTANT(CURLOPT_READFUNCTION);
+#if CURLOPT_PASSWDFUNCTION != 0 
 	REGISTER_CURL_CONSTANT(CURLOPT_PASSWDFUNCTION);
+#endif
 	REGISTER_CURL_CONSTANT(CURLOPT_HEADERFUNCTION);
 	REGISTER_CURL_CONSTANT(CURLOPT_MAXREDIRS);
@@ -442,4 +453,7 @@
 	REGISTER_CURL_CONSTANT(CURLOPT_UNRESTRICTED_AUTH);
 	REGISTER_CURL_CONSTANT(CURLOPT_FTP_USE_EPRT);
+#if LIBCURL_VERSION_NUM > 0x070b01 /* CURLOPT_TCP_NODELAY is available since curl 7.11.2 */
+	REGISTER_CURL_CONSTANT(CURLOPT_TCP_NODELAY);
+#endif
 	REGISTER_CURL_CONSTANT(CURLOPT_HTTP200ALIASES);
 	REGISTER_CURL_CONSTANT(CURL_TIMECOND_IFMODSINCE);
@@ -608,5 +622,5 @@
 
 #ifdef PHP_CURL_NEED_OPENSSL_TSL
-	{
+	if (!CRYPTO_get_id_callback()) {
 		int i, c = CRYPTO_num_locks();
 		
@@ -661,14 +675,9 @@
 	php_unregister_url_stream_wrapper("ldap" TSRMLS_CC);
 #endif
-#ifdef PHP_CURL_NEED_OPENSSL_TSL
-	/* ensure there are valid callbacks set */
-	CRYPTO_set_id_callback(php_curl_ssl_id);
-	CRYPTO_set_locking_callback(php_curl_ssl_lock);
-#endif
 	curl_global_cleanup();
 #ifdef PHP_CURL_NEED_OPENSSL_TSL
 	if (php_curl_openssl_tsl) {
 		int i, c = CRYPTO_num_locks();
-			
+		
 		CRYPTO_set_id_callback(NULL);
 		CRYPTO_set_locking_callback(NULL);
@@ -1075,5 +1084,5 @@
 	if (argc > 0) {
 		convert_to_string_ex(url);
-		PHP_CURL_CHECK_OPEN_BASEDIR(Z_STRVAL_PP(url), Z_STRLEN_PP(url));
+		PHP_CURL_CHECK_OPEN_BASEDIR(Z_STRVAL_PP(url), Z_STRLEN_PP(url), (void) NULL);
 	}
 
@@ -1158,5 +1167,5 @@
 	dupch->handlers->read->fp = ch->handlers->read->fp;
 	dupch->handlers->read->fd = ch->handlers->read->fd;
-
+#if CURLOPT_PASSWDDATA != 0
 	if (ch->handlers->passwd) {
 		zval_add_ref(&ch->handlers->passwd);
@@ -1164,4 +1173,5 @@
 		curl_easy_setopt(ch->cp, CURLOPT_PASSWDDATA, (void *) dupch);
 	}
+#endif
 	if (ch->handlers->write->func_name) {
 		zval_add_ref(&ch->handlers->write->func_name);
@@ -1257,4 +1267,7 @@
 		case CURLOPT_AUTOREFERER:
 		case CURLOPT_COOKIESESSION:
+#if LIBCURL_VERSION_NUM > 0x070b01 /* CURLOPT_TCP_NODELAY is available since curl 7.11.2 */
+		case CURLOPT_TCP_NODELAY:
+#endif
 			convert_to_long_ex(zvalue);
 			error = curl_easy_setopt(ch->cp, option, Z_LVAL_PP(zvalue));
@@ -1265,5 +1278,6 @@
 				if (Z_LVAL_PP(zvalue) != 0) {
 					php_error_docref(NULL TSRMLS_CC, E_WARNING, "CURLOPT_FOLLOWLOCATION cannot be activated when in safe_mode or an open_basedir is set");
-					RETURN_FALSE;
+					RETVAL_FALSE;
+					return 1;
 				}
 			}
@@ -1298,5 +1312,5 @@
 
 			if (option == CURLOPT_URL) {
-				PHP_CURL_CHECK_OPEN_BASEDIR(Z_STRVAL_PP(zvalue), Z_STRLEN_PP(zvalue));
+				PHP_CURL_CHECK_OPEN_BASEDIR(Z_STRVAL_PP(zvalue), Z_STRLEN_PP(zvalue), 1);
 			}
 
@@ -1316,5 +1330,8 @@
 		
 			what = zend_fetch_resource(zvalue TSRMLS_CC, -1, "File-Handle", &type, 1, php_file_le_stream());
-			ZEND_VERIFY_RESOURCE(what);
+			if (!what) {
+				RETVAL_FALSE;
+				return 1;
+			}
 
 			if (FAILURE == php_stream_cast((php_stream *) what, PHP_STREAM_AS_STDIO, (void *) &fp, REPORT_ERRORS)) {
