diff -r -U2 kde-workspace-4.11.0/CMakeLists.txt kde-workspace-4.11.0/CMakeLists.txt
--- kde-workspace-4.11.0/CMakeLists.txt	2013-08-06 14:49:30.000000000 -0700
+++ kde-workspace-4.11.0/CMakeLists.txt	2013-08-06 14:49:30.000000000 -0700
@@ -25,5 +25,5 @@
 set_package_properties(Strigi PROPERTIES DESCRIPTION "Desktop indexing and search support"
                        URL "http://strigi.sourceforge.net"
-                       TYPE REQUIRED
+                       TYPE OPTIONAL
                       )
 
diff -r -U2 kde-workspace-4.11.0/kcontrol/kfontinst/CMakeLists.txt kde-workspace-4.11.0/kcontrol/kfontinst/CMakeLists.txt
--- kde-workspace-4.11.0/kcontrol/kfontinst/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
+++ kde-workspace-4.11.0/kcontrol/kfontinst/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
@@ -5,5 +5,5 @@
     find_package(Freetype REQUIRED)
     find_package(Fontconfig REQUIRED)
-    find_package(Strigi REQUIRED)
+    find_package(Strigi OPTIONAL)
     find_package(X11 REQUIRED)
     macro_log_feature(X11_Xft_FOUND "libxft" "X FreeType interface library" "http://www.x.org" FALSE "" "Font installer and font previews")
@@ -41,5 +41,4 @@
     add_subdirectory( dbus )
     add_subdirectory( kcmfontinst ) 
-    add_subdirectory( strigi-analyzer )
     add_subdirectory( apps )
     add_subdirectory( kio ) 
diff -r -U2 kde-workspace-4.11.0/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt kde-workspace-4.11.0/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt
--- kde-workspace-4.11.0/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
+++ kde-workspace-4.11.0/kcontrol/kfontinst/strigi-analyzer/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
@@ -1,4 +1,3 @@
 include_directories( ${KDEBASE_WORKSPACE_SOURCE_DIR}/kcontrol/kfontinst/lib/
-                     ${STRIGI_INCLUDE_DIR}
                    )
 
diff -r -U2 kde-workspace-4.11.0/plasma/desktop/applets/kickoff/CMakeLists.txt kde-workspace-4.11.0/plasma/desktop/applets/kickoff/CMakeLists.txt
--- kde-workspace-4.11.0/plasma/desktop/applets/kickoff/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
+++ kde-workspace-4.11.0/plasma/desktop/applets/kickoff/CMakeLists.txt	2013-07-09 15:44:07.000000000 -0700
@@ -39,6 +39,4 @@
 endif(HAVE_STRIGIDBUS)
 
-include_directories( ${STRIGI_INCLUDE_DIR} )
-
 kde4_add_library(kickoff SHARED ${libkickoff_SRCS})
 target_link_libraries(kickoff ${KDE4_PLASMA_LIBS} ${Kickoff_LIBS})
