diff -N -r -U2 glibc-2.7/sysdeps/i386/i586/memcpy_chk.S glibc-2.7/sysdeps/i386/i586/memcpy_chk.S
--- glibc-2.7/sysdeps/i386/i586/memcpy_chk.S	2008-05-04 01:03:26.000000000 -0700
+++ glibc-2.7/sysdeps/i386/i586/memcpy_chk.S	2008-05-04 01:03:26.000000000 -0700
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memcpy_chk.S>
diff -N -r -U2 glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S
--- glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S	2008-05-04 01:03:26.000000000 -0700
+++ glibc-2.7/sysdeps/i386/i586/mempcpy_chk.S	2008-05-04 01:03:26.000000000 -0700
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/mempcpy_chk.S>
diff -N -r -U2 glibc-2.7/sysdeps/i386/i586/memset_chk.S glibc-2.7/sysdeps/i386/i586/memset_chk.S
--- glibc-2.7/sysdeps/i386/i586/memset_chk.S	2008-05-04 01:03:26.000000000 -0700
+++ glibc-2.7/sysdeps/i386/i586/memset_chk.S	2008-05-04 01:03:26.000000000 -0700
@@ -0,0 +1 @@
+#include <sysdeps/i386/i686/memset_chk.S>
