diff -r -U2 gcc-4.2.1/gcc/Makefile.in gcc-4.2.1/gcc/Makefile.in
--- gcc-4.2.1/gcc/Makefile.in	2007-05-31 08:37:38.000000000 -0700
+++ gcc-4.2.1/gcc/Makefile.in	2007-05-31 08:37:38.000000000 -0700
@@ -3255,7 +3255,5 @@
 	    SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
 	    export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	    cd $(build_objdir)/fixincludes && \
-	    $(SHELL) ./fixinc.sh ../../gcc/include \
-	      $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
+	    cd $(build_objdir)/fixincludes ); \
 	  rm -f include/syslimits.h; \
 	  if [ -f include/limits.h ]; then \
