--- apmd/Makefile.ori	2002-06-03 14:58:17.000000000 -0700
+++ apmd/Makefile	2002-06-03 14:58:56.000000000 -0700
@@ -29,5 +29,5 @@
 
 OBJS=apmlib.o
-EXES=apm apmd xapm apmsleep
+EXES=apm apmd apmsleep
 HEADERS=apm.h
 
@@ -63,12 +63,6 @@
 	$(CC) -o $@ $< $(LDFLAGS) $(XTRALDFLAGS)
 
-xapm.o: xapm.c
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) $(XTRACFLAGS) -DNARROWPROTO $<
-
 apmsleep: apmsleep.o
 
-xapm: xapm.o $(LIB)
-	$(CC) -o $@ $< $(LDFLAGS) $(XTRALDFLAGS) $(XLDFLAGS) $(XLIBS)
-
 libapm.a: $(OBJS)
 	$(AR) $(ARFLAGS) $@ $(OBJS)
@@ -95,6 +89,4 @@
 	install -b -Vt apmd_proxy      ${APMD_PROXY_DIR}
 	install -m 644 apmd.8          ${MANDIR}/man8
-	install        xapm            ${PREFIX}/bin
-	install -m 644 xapm.1          ${MANDIR}/man1/xapm.1
 	install        apmsleep	       ${PREFIX}/bin
 	install -m 644 apmsleep.1      ${MANDIR}/man1
