diff -r -U2 prboom-plus-2.5.0.3/config.h prboom-plus-2.5.0.3/config.h
--- prboom-plus-2.5.0.3/config.h	2009-06-30 02:14:06.000000000 -0700
+++ prboom-plus-2.5.0.3/config.h	2009-06-30 02:14:06.000000000 -0700
@@ -10,5 +10,5 @@
 
 /* Define to be the path where Doom WADs are stored */
-#define DOOMWADDIR "/usr/local/share/games/doom"
+#define DOOMWADDIR "/usr/share/games/doom"
 
 /* Define if you are building with OpenGL support */
