diff --git a/src/xdrv/wcmCommon.c b/src/xdrv/wcmCommon.c
index 71b69df..5806d18 100755
--- a/src/xdrv/wcmCommon.c
+++ b/src/xdrv/wcmCommon.c
@@ -1091,21 +1091,6 @@ void xf86WcmEvent(WacomCommonPtr common, unsigned int channel,
 
 	DBG(10, common->debugLevel, ErrorF("xf86WcmEvent at channel = %d\n", channel));
 
-	/* Tool on the tablet when driver starts. This sometime causes
-	 * access errors to the device */
-#if defined WCM_XFREE86 || GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
-	if (!miPointerCurrentScreen())
-#else
-	if (!miPointerGetScreen(pDev->dev))
-#endif
-	{
-		DBG(1, common->debugLevel, ErrorF("xf86WcmEvent: "
-			"Wacom driver can not get Current Screen ID\n"));
-		DBG(1, common->debugLevel, ErrorF(
-			"Please remove Wacom tool from the tablet.\n"));
-		return;
-	}
-
 	/* sanity check the channel */
 	if (channel >= MAX_CHANNELS)
 		return;
@@ -1363,6 +1348,21 @@ static void commonDispatchDevice(WacomCommonPtr common, unsigned int channel,
 	}
 	/* X: InputDevice selection done! */
 
+	/* Tool on the tablet when driver starts. This sometime causes
+	 * access errors to the device */
+#if defined WCM_XFREE86 || GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 0
+	if (!miPointerCurrentScreen())
+#else
+	if (!miPointerGetScreen(pDev->dev))
+#endif
+	{
+		DBG(1, common->debugLevel, ErrorF("xf86WcmEvent: "
+			"Wacom driver can not get Current Screen ID\n"));
+		DBG(1, common->debugLevel, ErrorF(
+			"Please remove Wacom tool from the tablet.\n"));
+		return;
+	}
+
 	/* if a device matched criteria, handle filtering per device
 	 * settings, and send event to XInput */
 	if (pDev)
