diff -r -U2 SDL-1.2.6/src/video/SDL_blit_A.c SDL-1.2.6/src/video/SDL_blit_A.c
--- SDL-1.2.6/src/video/SDL_blit_A.c	2003-08-30 12:13:06.000000000 -0700
+++ SDL-1.2.6/src/video/SDL_blit_A.c	2003-08-30 12:13:06.000000000 -0700
@@ -221,9 +221,9 @@
         Uint8 load[8];
   
-        *(Uint64 *)load = 0x00fefefe00fefefe;/* alpha128 mask */
+        *(Uint64 *)load = 0x00fefefe00fefefeULL;/* alpha128 mask */
         movq_m2r(*load, mm4); /* alpha128 mask -> mm4 */
-        *(Uint64 *)load = 0x0001010100010101;/* !alpha128 mask */
+        *(Uint64 *)load = 0x0001010100010101ULL;/* !alpha128 mask */
         movq_m2r(*load, mm3); /* !alpha128 mask -> mm3 */
-        *(Uint64 *)load = 0xFF000000FF000000;/* dst alpha mask */
+        *(Uint64 *)load = 0xFF000000FF000000ULL;/* dst alpha mask */
         movq_m2r(*load, mm7); /* dst alpha mask -> mm7 */
 	while(height--) {
@@ -277,8 +277,8 @@
 					
                 movq_m2r(*load, mm4); /* alpha -> mm4 */
-		*(Uint64 *)load = 0x00FF00FF00FF00FF;
+		*(Uint64 *)load = 0x00FF00FF00FF00FFULL;
                 movq_m2r(*load, mm3); /* mask -> mm3 */
 		pand_r2r(mm3, mm4); /* mm4 & mask -> 0A0A0A0A -> mm4 */
-		*(Uint64 *)load = 0xFF000000FF000000;/* dst alpha mask */
+		*(Uint64 *)load = 0xFF000000FF000000ULL;/* dst alpha mask */
 		movq_m2r(*load, mm7); /* dst alpha mask -> mm7 */
 		
@@ -361,11 +361,11 @@
         Uint8 load[8];
 	                
-	*(Uint64 *)load = 0x00FF00FF00FF00FF;
+	*(Uint64 *)load = 0x00FF00FF00FF00FFULL;
         movq_m2r(*load, mm3); /* mask -> mm2 */
-	*(Uint64 *)load = 0x00FF000000000000;
+	*(Uint64 *)load = 0x00FF000000000000ULL;
         movq_m2r(*load, mm7); /* dst alpha mask -> mm2 */
-        *(Uint64 *)load = 0x00FFFFFF00FFFFFF;
+        *(Uint64 *)load = 0x00FFFFFF00FFFFFFULL;
         movq_m2r(*load, mm0); /* alpha 255 mask -> mm0 */
-        *(Uint64 *)load = 0xFF000000FF000000;
+        *(Uint64 *)load = 0xFF000000FF000000ULL;
         movq_m2r(*load, mm6); /* alpha 255 !mask -> mm6 */
 	while(height--) {
@@ -793,9 +793,9 @@
 	  
  	        /* Setup the 565 color channel masks */
-	        *(Uint64 *)load = 0xF800F800F800F800;
+	        *(Uint64 *)load = 0xF800F800F800F800ULL;
 		movq_m2r(*load, mm1); /* MASKRED -> mm1 */
-		*(Uint64 *)load = 0x07E007E007E007E0;
+		*(Uint64 *)load = 0x07E007E007E007E0ULL;
 		movq_m2r(*load, mm4); /* MASKGREEN -> mm4 */
-		*(Uint64 *)load = 0x001F001F001F001F;
+		*(Uint64 *)load = 0x001F001F001F001FULL;
 		movq_m2r(*load, mm7); /* MASKBLUE -> mm7 */
 		while(height--) {
@@ -943,9 +943,9 @@
 	  
  	        /* Setup the 555 color channel masks */
-	        *(Uint64 *)load = 0x7C007C007C007C00;
+	        *(Uint64 *)load = 0x7C007C007C007C00ULL;
 		movq_m2r(*load, mm1); /* MASKRED -> mm1 */
-		*(Uint64 *)load = 0x03E003E003E003E0;
+		*(Uint64 *)load = 0x03E003E003E003E0ULL;
 		movq_m2r(*load, mm4); /* MASKGREEN -> mm4 */
-		*(Uint64 *)load = 0x001F001F001F001F;
+		*(Uint64 *)load = 0x001F001F001F001FULL;
 		movq_m2r(*load, mm7); /* MASKBLUE -> mm7 */
 		while(height--) {
diff -r -U2 SDL-1.2.6/src/video/mmx.h SDL-1.2.6/src/video/mmx.h
--- SDL-1.2.6/src/video/mmx.h	2003-08-21 22:51:19.000000000 -0700
+++ SDL-1.2.6/src/video/mmx.h	2003-08-21 22:51:19.000000000 -0700
@@ -334,6 +334,6 @@
 				      #op " %1, %%mm0\n\t" \
 				      "movq %%mm0, %0" \
-				      : "=X" (memd) \
-				      : "X" (mems)); \
+				      : "=m" (memd) \
+				      : "m" (mems)); \
 		mmx_trace = (memd); \
 		printf(#memd "=0x%08x%08x\n", \
@@ -349,14 +349,14 @@
 	__asm__ __volatile__ (#op " %0, %%" #reg \
 			      : /* nothing */ \
-			      : "X" (imm) )
+			      : "i" (imm) )
 
 #define	mmx_m2r(op, mem, reg) \
 	__asm__ __volatile__ (#op " %0, %%" #reg \
 			      : /* nothing */ \
-			      : "X" (mem))
+			      : "m" (mem))
 
 #define	mmx_r2m(op, reg, mem) \
 	__asm__ __volatile__ (#op " %%" #reg ", %0" \
-			      : "=X" (mem) \
+			      : "=m" (mem) \
 			      : /* nothing */ )
 
