diff -r -U2 gimp-2.6.6/app/sanity.c gimp-2.6.6/app/sanity.c
--- gimp-2.6.6/app/sanity.c	2008-11-20 15:43:08.000000000 -0700
+++ gimp-2.6.6/app/sanity.c	2008-11-20 15:43:08.000000000 -0700
@@ -61,4 +61,5 @@
     abort_message = sanity_check_freetype ();
 
+/*
   if (! abort_message)
     abort_message = sanity_check_babl ();
@@ -66,4 +67,5 @@
   if (! abort_message)
     abort_message = sanity_check_gegl ();
+*/
 
   if (! abort_message)
