diff -r -U2 syslinux-2.06/Makefile syslinux-2.06/Makefile
--- syslinux-2.06/Makefile	2003-07-15 00:49:04.000000000 -0700
+++ syslinux-2.06/Makefile	2003-07-15 00:49:04.000000000 -0700
@@ -56,5 +56,5 @@
 BTARGET = kwdhash.gen version.gen ldlinux.bss ldlinux.sys ldlinux.bin \
 	  pxelinux.0 mbr.bin isolinux.bin isolinux-debug.bin \
-	  libsyslinux.a syslinux.exe $(LIB_SO) 
+	  libsyslinux.a $(LIB_SO) 
 ITARGET = syslinux.com syslinux syslinux-nomtools copybs.com gethostip \
 	  mkdiskimage
@@ -69,5 +69,5 @@
 # Things to install in /usr/lib/syslinux
 INSTALL_AUX   =	pxelinux.0 isolinux.bin isolinux-debug.bin \
-		syslinux.com syslinux.exe copybs.com memdisk/memdisk
+		syslinux.com copybs.com memdisk/memdisk
 # Things to install in /usr/lib
 INSTALL_LIB   = $(LIB_SO) libsyslinux.a
@@ -200,5 +200,4 @@
 tidy: local-tidy
 	$(MAKE) -C memdisk tidy
-	$(MAKE) -C win32 tidy
 
 local-clean:
@@ -208,5 +207,4 @@
 	$(MAKE) -C sample clean
 	$(MAKE) -C memdisk clean
-	$(MAKE) -C win32 clean
 
 dist: tidy
@@ -221,5 +219,4 @@
 	$(MAKE) -C sample spotless
 	$(MAKE) -C memdisk spotless
-	$(MAKE) -C win32 spotless
 
 .depend:
diff -r -U2 syslinux-2.06/memdisk/init.S16 syslinux-2.06/memdisk/init.S16
--- syslinux-2.06/memdisk/init.S16	2003-04-09 23:28:08.000000000 -0700
+++ syslinux-2.06/memdisk/init.S16	2003-04-09 23:28:08.000000000 -0700
@@ -43,5 +43,4 @@
 		jmp	start
 
-# This is the setup header, and it must start at %cs:2 (old 0x9020:2)
 	
 		.ascii	"HdrS"		# header signature
@@ -58,5 +57,4 @@
 					# assigned ids
 	
-# flags, unused bits must be zero (RFU) bit within loadflags
 loadflags:
 LOADED_HIGH	= 1			# If set, the kernel is loaded high
