diff -r -U2 glibc-2.7/configure glibc-2.7/configure
--- glibc-2.7/configure	2007-10-18 04:22:23.000000000 -0700
+++ glibc-2.7/configure	2007-10-18 04:22:23.000000000 -0700
@@ -5064,5 +5064,5 @@
 if test -n "$sysheaders"; then
   ccheaders=`$CC -print-file-name=include`
-  SYSINCLUDES="-nostdinc -isystem $ccheaders \
+  SYSINCLUDES="-nostdinc -isystem $ccheaders -isystem $ccheaders-fixed \
 -isystem `echo $sysheaders | sed 's/:/ -isystem /g'`"
   if test -n "$CXX"; then
