diff -r -U2 gcc-4.3.0/gcc/Makefile.in gcc-4.3.0/gcc/Makefile.in
--- gcc-4.3.0/gcc/Makefile.in	2008-02-25 07:53:34.000000000 -0700
+++ gcc-4.3.0/gcc/Makefile.in	2008-02-25 07:53:34.000000000 -0700
@@ -3509,7 +3509,5 @@
 	      SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
 	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh ../../gcc/$${fix_dir} \
-	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
+	      cd $(build_objdir)/fixincludes ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
