diff -r -U2 sdlmame0136/src/osd/sdl/dview.c sdlmame0136/src/osd/sdl/dview.c
--- sdlmame0136/src/osd/sdl/dview.c	2009-12-19 09:18:15.000000000 -0700
+++ sdlmame0136/src/osd/sdl/dview.c	2009-12-19 09:18:15.000000000 -0700
@@ -214,5 +214,5 @@
 	debug_view_set_visible_size(dv->dw, vsize);
 
-	if(GTK_WIDGET_REALIZED(wdv))
+	if(gtk_widget_get_realized(wdv))
 		gdk_window_move_resize(wdv->window,
 							   allocation->x, allocation->y,
