diff -uNr kdepim-4.4.5.old//kleopatra/crypto/gui/newresultpage.cpp kdepim-4.4.5/kleopatra/crypto/gui/newresultpage.cpp
--- kdepim-4.4.5.old//kleopatra/crypto/gui/newresultpage.cpp	2011-01-13 09:53:12.892928551 -0500
+++ kdepim-4.4.5/kleopatra/crypto/gui/newresultpage.cpp	2011-01-13 09:53:48.912678586 -0500
@@ -30,6 +30,8 @@
     your version.
 */
 
+#include <boost/mem_fn.hpp>
+
 #include <config-kleopatra.h>
 
 #include "newresultpage.h"
