diff -r -U2 gnome-print-0.37/installer/gf-pfb.c gnome-print-0.37/installer/gf-pfb.c
--- gnome-print-0.37/installer/gf-pfb.c	2002-01-18 17:23:00.000000000 -0700
+++ gnome-print-0.37/installer/gf-pfb.c	2002-01-18 17:23:00.000000000 -0700
@@ -13,5 +13,6 @@
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include "gf-pfb.h"
 
diff -r -U2 gnome-print-0.37/installer/gf-ttf.c gnome-print-0.37/installer/gf-ttf.c
--- gnome-print-0.37/installer/gf-ttf.c	2001-11-05 07:07:28.000000000 -0700
+++ gnome-print-0.37/installer/gf-ttf.c	2001-11-05 07:07:28.000000000 -0700
@@ -33,5 +33,6 @@
 #include <string.h>
 #include <ctype.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 #include "gf-ttf.h"
diff -r -U2 gnome-print-0.37/installer/gnome-font-install.c gnome-print-0.37/installer/gnome-font-install.c
--- gnome-print-0.37/installer/gnome-font-install.c	2002-01-22 14:44:55.000000000 -0700
+++ gnome-print-0.37/installer/gnome-font-install.c	2002-01-22 14:44:55.000000000 -0700
@@ -27,5 +27,6 @@
 #include <xmlmemory.h>
 /* End of ugly thing */
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnome/gnome-defs.h>
 #include <libgnome/gnome-util.h>
diff -r -U2 gnome-print-0.37/libgnomeprint/gnome-font-private.h gnome-print-0.37/libgnomeprint/gnome-font-private.h
--- gnome-print-0.37/libgnomeprint/gnome-font-private.h	2001-12-05 15:07:58.000000000 -0700
+++ gnome-print-0.37/libgnomeprint/gnome-font-private.h	2001-12-05 15:07:58.000000000 -0700
@@ -23,5 +23,6 @@
 BEGIN_GNOME_DECLS
 
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <libgnomeprint/gp-character-block.h>
 #include <libgnomeprint/gp-fontmap.h>
diff -r -U2 gnome-print-0.37/libgnomeprint/parseTT.h gnome-print-0.37/libgnomeprint/parseTT.h
--- gnome-print-0.37/libgnomeprint/parseTT.h	2001-11-05 12:27:09.000000000 -0700
+++ gnome-print-0.37/libgnomeprint/parseTT.h	2001-11-05 12:27:09.000000000 -0700
@@ -23,5 +23,6 @@
 
 #include <glib.h>
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 /*
