diff -r -U2 tcp_wrappers_7.6/Makefile tcp_wrappers_7.6/Makefile
--- tcp_wrappers_7.6/Makefile	1997-03-21 11:27:21.000000000 -0700
+++ tcp_wrappers_7.6/Makefile	1997-03-21 11:27:21.000000000 -0700
@@ -45,5 +45,5 @@
 #
 # SysV.4 Solaris 2.x OSF AIX
-#REAL_DAEMON_DIR=/usr/sbin
+REAL_DAEMON_DIR=/usr/sbin
 #
 # BSD 4.4
@@ -144,5 +144,5 @@
 linux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP= TLI= EXTRA_CFLAGS="-DBROKEN_SO_LINGER" all
 
@@ -392,5 +392,5 @@
 # implements setenv(), getenv(), and putenv().
 
-AUX_OBJ= setenv.o
+#AUX_OBJ= setenv.o
 #AUX_OBJ= environ.o
 #AUX_OBJ= environ.o strcasecmp.o
@@ -658,5 +658,5 @@
 
 SHELL	= /bin/sh
-.c.o:;	$(CC) $(CFLAGS) -c $*.c
+.c.o:;	$(LIBTOOL) $(CC) $(CFLAGS) -c $*.c
 
 CFLAGS	= -O -DFACILITY=$(FACILITY) $(ACCESS) $(PARANOID) $(NETGROUP) \
@@ -705,24 +705,24 @@
 
 tcpd:	tcpd.o $(LIB)
-	$(CC) $(CFLAGS) -o $@ tcpd.o $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ tcpd.o $(LIB) $(LIBS)
 
 miscd:	miscd.o $(LIB)
-	$(CC) $(CFLAGS) -o $@ miscd.o $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ miscd.o $(LIB) $(LIBS)
 
 safe_finger: safe_finger.o $(LIB)
-	$(CC) $(CFLAGS) -o $@ safe_finger.o $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ safe_finger.o $(LIB) $(LIBS)
 
 TCPDMATCH_OBJ = tcpdmatch.o fakelog.o inetcf.o scaffold.o
 
 tcpdmatch: $(TCPDMATCH_OBJ) $(LIB)
-	$(CC) $(CFLAGS) -o $@ $(TCPDMATCH_OBJ) $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ $(TCPDMATCH_OBJ) $(LIB) $(LIBS)
 
 try-from: try-from.o fakelog.o $(LIB)
-	$(CC) $(CFLAGS) -o $@ try-from.o fakelog.o $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ try-from.o fakelog.o $(LIB) $(LIBS)
 
 TCPDCHK_OBJ = tcpdchk.o fakelog.o inetcf.o scaffold.o
 
 tcpdchk: $(TCPDCHK_OBJ) $(LIB)
-	$(CC) $(CFLAGS) -o $@ $(TCPDCHK_OBJ) $(LIB) $(LIBS)
+	$(LIBTOOL) $(CC) $(CFLAGS) -o $@ $(TCPDCHK_OBJ) $(LIB) $(LIBS)
 
 shar:	$(KIT)
diff -r -U2 tcp_wrappers_7.6/environ.c tcp_wrappers_7.6/environ.c
--- tcp_wrappers_7.6/environ.c	1994-03-23 08:09:48.000000000 -0700
+++ tcp_wrappers_7.6/environ.c	1994-03-23 08:09:48.000000000 -0700
@@ -14,11 +14,6 @@
 
 extern char **environ;
-extern char *strchr();
-extern char *strcpy();
-extern char *strncpy();
-extern char *malloc();
-extern char *realloc();
-extern int strncmp();
-extern void free();
+#include <stdlib.h>
+#include <string.h>
 
 #ifdef no_memcpy
diff -r -U2 tcp_wrappers_7.6/percent_m.c tcp_wrappers_7.6/percent_m.c
--- tcp_wrappers_7.6/percent_m.c	1994-12-28 09:42:37.000000000 -0700
+++ tcp_wrappers_7.6/percent_m.c	1994-12-28 09:42:37.000000000 -0700
@@ -14,8 +14,4 @@
 
 extern int errno;
-#ifndef SYS_ERRLIST_DEFINED
-extern char *sys_errlist[];
-extern int sys_nerr;
-#endif
 
 #include "mystdarg.h"
diff -r -U2 tcp_wrappers_7.6/scaffold.c tcp_wrappers_7.6/scaffold.c
--- tcp_wrappers_7.6/scaffold.c	1997-03-21 11:27:24.000000000 -0700
+++ tcp_wrappers_7.6/scaffold.c	2004-10-18 15:16:54.539165960 -0700
@@ -26,5 +26,5 @@
 #endif
 
-extern char *malloc();
+#include <stdlib.h>
 
 /* Application-specific. */
