diff -r -U2 csmash-0.6.6/BallView.h csmash-0.6.6/BallView.h
--- csmash-0.6.6/BallView.h	2003-07-13 08:26:24.000000000 -0700
+++ csmash-0.6.6/BallView.h	2003-07-13 08:26:24.000000000 -0700
@@ -33,5 +33,5 @@
   static GLuint m_number[10];
 private:
-  void BallView::DrawTargetCircle();
+  void DrawTargetCircle();
   GLUquadricObj *m_quad;
 
diff -r -U2 csmash-0.6.6/loadparts.cpp csmash-0.6.6/loadparts.cpp
--- csmash-0.6.6/loadparts.cpp	2003-11-19 09:49:31.000000000 -0700
+++ csmash-0.6.6/loadparts.cpp	2003-11-19 09:49:31.000000000 -0700
@@ -71,6 +71,6 @@
 
 template <typename T>
-inline const T& clamp(const T& a, const T& x, const T& b) {
-    if (a > x) return a;
+inline const T& clamp(const T& x, const T& b) {
+    if (0 > x) return 0;
     elif (b < x) return b;
     else return x;
@@ -246,5 +246,5 @@
 	while ('\\' == line[l-1]) {
             // concat next line(s)
-	    int bufsize = clamp(0U, sizeof(line)-l, sizeof(line)-1);
+	    int bufsize = clamp(sizeof(line)-l, sizeof(line)-1);
 	    fgets(&line[l-2], bufsize, fp);
 	    if (feof((FILE*)fp)) break;
diff -r -U2 csmash-0.6.6/matrix csmash-0.6.6/matrix
--- csmash-0.6.6/matrix	2002-03-06 09:46:01.000000000 -0700
+++ csmash-0.6.6/matrix	2002-03-06 09:46:01.000000000 -0700
@@ -43,5 +43,4 @@
 /* __BEGIN__BEGIN__ */
 //__NAMESPACE_BEGIN(ese);
-#if (!defined __GNUG__) || (__GNUC__ == 3)
 template<typename T>
 void swap(T& a, T& b)
@@ -51,5 +50,4 @@
     b = c;
 }
-#endif
 
 // matrix_size MUST BE larger than 1
Only in fixed: csmash-0.6.6.patch
