diff -r -U2 procps-3.2.1/ps/parser.c procps-3.2.1/ps/parser.c
--- procps-3.2.1/ps/parser.c	2004-04-19 23:03:06.667624800 -0700
+++ procps-3.2.1/ps/parser.c	2004-04-19 23:03:06.667624800 -0700
@@ -568,8 +568,4 @@
       format_modifiers |= FM_M;
       break;
-    case 'Z':
-      trace("SElinux Security Context format\n");
-      format_flags |= FF_Fc;
-      break;
     case 'a':
       trace("a Select all w/tty, including other users\n");
