diff -r -U2 postgresql-9.0.4/src/pl/plperl/plperl.c postgresql-9.0.4/src/pl/plperl/plperl.c
--- postgresql-9.0.4/src/pl/plperl/plperl.c	2011-04-14 20:15:53.000000000 -0700
+++ postgresql-9.0.4/src/pl/plperl/plperl.c	2011-04-14 20:15:53.000000000 -0700
@@ -875,5 +875,5 @@
 			continue;
 		SvREFCNT_dec(GvCV(sv)); /* free the CV */
-		GvCV(sv) = NULL;		/* prevent call via GV */
+		GvCV_set(sv,NULL);		/* prevent call via GV */
 	}
 	hv_clear(stash);
