diff -r -U2 gcc-4.7.1/gcc/Makefile.in gcc-4.7.1/gcc/Makefile.in
--- gcc-4.7.1/gcc/Makefile.in	2012-05-18 09:32:05.000000000 -0700
+++ gcc-4.7.1/gcc/Makefile.in	2012-05-18 09:32:05.000000000 -0700
@@ -4184,11 +4184,4 @@
 	    $(mkinstalldirs) $${fix_dir}; \
 	    chmod a+rx $${fix_dir} || true; \
-	    (TARGET_MACHINE='$(target)'; srcdir=`cd $(srcdir); ${PWD_COMMAND}`; \
-	      SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
-	      gcc_dir=`${PWD_COMMAND}` ; \
-	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh "$${gcc_dir}/$${fix_dir}" \
-	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
Only in gcc-4.7.1/gcc: Makefile.in.orig
