diff -r -U2 zlib-1.2.3/configure zlib-1.2.3/configure
--- zlib-1.2.3/configure	2005-07-11 13:11:57.000000000 -0700
+++ zlib-1.2.3/configure	2005-07-11 13:11:57.000000000 -0700
@@ -172,16 +172,8 @@
   echo Checking for shared library support...
   # we must test in two steps (cc then ld), required at least on SunOS 4.x
-  if test "`($CC -c $SFLAGS $test.c) 2>&1`" = "" &&
-     test "`($LDSHARED -o $test$shared_ext $test.o) 2>&1`" = ""; then
     CFLAGS="$SFLAGS"
     LIBS="$SHAREDLIBV"
+    LDSHARED="$LDSHARED $SFLAGS"
     echo Building shared library $SHAREDLIBV with $CC.
-  elif test -z "$old_cc" -a -z "$old_cflags"; then
-    echo No shared library support.
-    shared=0;
-  else
-    echo 'No shared library support; try without defining CC and CFLAGS'
-    shared=0;
-  fi
 fi
 if test $shared -eq 0; then
