diff -r -U2 kdepim-4.10.4/CMakeLists.txt kdepim-4.10.4/CMakeLists.txt
--- kdepim-4.10.4/CMakeLists.txt	2013-05-28 12:24:34.000000000 -0700
+++ kdepim-4.10.4/CMakeLists.txt	2013-05-28 12:24:34.000000000 -0700
@@ -194,7 +194,4 @@
   macro_log_feature(ZLIB_FOUND "ZLib" "The Zlib compression library" "http://www.zlib.net" TRUE "" "")
 
-  find_package(Strigi)
-  macro_log_feature(STRIGI_FOUND "Strigi" "Index metadata of files" "http://strigi.sourceforge.net" TRUE "" "")
-
   find_package(QGpgme)
   macro_log_feature(QGPGME_FOUND "QGpgME" "The QGpgME library" "http://www.kde.org" FALSE "" "QGpgME is required to build KMail, KOrganizer and Kleopatra")
@@ -378,5 +375,4 @@
   macro_optional_add_subdirectory(akonadiconsole)
   macro_optional_add_subdirectory(console)
-  macro_optional_add_subdirectory(strigi-analyzer)
 
   # These targets depend on optional applications
