diff -r -U2 kdelibs-4.10.4/CMakeLists.txt kdelibs-4.10.4/CMakeLists.txt
--- kdelibs-4.10.4/CMakeLists.txt	2013-05-30 12:46:37.000000000 -0700
+++ kdelibs-4.10.4/CMakeLists.txt	2013-05-30 12:46:37.000000000 -0700
@@ -80,5 +80,5 @@
     set (STRIGI_EXTRA_TEXT "")
 else (WIN32)
-    set (STRIGI_REQUIRED TRUE)
+    set (STRIGI_REQUIRED FALSE)
     set (STRIGI_EXTRA_TEXT "Required by some critical kioslaves")
 endif (WIN32)
diff -r -U2 kdelibs-4.10.4/kio/kio/kfilemetainfo.cpp kdelibs-4.10.4/kio/kio/kfilemetainfo.cpp
--- kdelibs-4.10.4/kio/kio/kfilemetainfo.cpp	2013-05-28 11:32:49.000000000 -0700
+++ kdelibs-4.10.4/kio/kio/kfilemetainfo.cpp	2013-05-28 11:32:49.000000000 -0700
@@ -422,7 +422,7 @@
 }
 
-const KFileMetaInfo& KFileMetaInfo::operator= ( KFileMetaInfo const & kfmi )
+KFileMetaInfo& KFileMetaInfo::operator= ( KFileMetaInfo const & kfmi )
 {
-    return kfmi;
+    return *this;
 }
 
