diff -r -U2 mjpegtools-1.8.0/lavtools/lav_io.c mjpegtools-1.8.0/lavtools/lav_io.c
--- mjpegtools-1.8.0/lavtools/lav_io.c	2005-08-27 04:47:20.000000000 -0700
+++ mjpegtools-1.8.0/lavtools/lav_io.c	2005-08-27 04:47:20.000000000 -0700
@@ -1368,5 +1368,5 @@
 #ifdef HAVE_LIBQUICKTIME
       case 'q':
-         res = fileno(((quicktime_t *)lav_file->qt_fd)->stream);
+         res = lqt_fileno((quicktime_t *)lav_file->qt_fd);
          break;
 #endif
diff -r -U2 mjpegtools-1.8.0/y4mdenoise/Region2D.hh mjpegtools-1.8.0/y4mdenoise/Region2D.hh
--- mjpegtools-1.8.0/y4mdenoise/Region2D.hh	2004-07-17 23:43:01.000000000 -0700
+++ mjpegtools-1.8.0/y4mdenoise/Region2D.hh	2004-07-17 23:43:01.000000000 -0700
@@ -17,4 +17,5 @@
 #include "Status_t.h"
 #include <iostream>
+#include <cassert>
 
 
