diff -r -U2 mozilla-2.0/toolkit/system/gnome/nsAlertsIconListener.cpp mozilla-2.0/toolkit/system/gnome/nsAlertsIconListener.cpp
--- mozilla-2.0/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-18 16:34:12.000000000 -0700
+++ mozilla-2.0/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-18 16:34:12.000000000 -0700
@@ -224,10 +224,5 @@
   mNotification = notify_notification_new(mAlertTitle.get(),
                                           mAlertText.get(),
-                                          NULL
-// >=libnotify-0.7.0 has no support for attaching to widgets
-#if !NOTIFY_CHECK_VERSION(0,7,0)
-                                          , NULL
-#endif
-                                          );
+                                          NULL );
 
   if (!mNotification)
