diff -r -U2 kde-baseapps-4.10.4/CMakeLists.txt kde-baseapps-4.10.4/CMakeLists.txt
--- kde-baseapps-4.10.4/CMakeLists.txt	2013-05-28 11:41:11.000000000 -0700
+++ kde-baseapps-4.10.4/CMakeLists.txt	2013-05-28 11:41:11.000000000 -0700
@@ -11,5 +11,5 @@
 set(SOPRANO_MIN_VERSION "2.7.56")
 find_package(KDE4 4.8.0 REQUIRED)
-find_package(Strigi REQUIRED)
+find_package(Strigi)
 find_package(ZLIB REQUIRED)
 include (KDE4Defaults)
