diff -r -U2 glibc-2.3/Makefile glibc-2.3/Makefile
--- glibc-2.3/Makefile	2002-09-14 19:18:51.000000000 -0700
+++ glibc-2.3/Makefile	2002-09-14 19:18:51.000000000 -0700
@@ -100,27 +100,4 @@
 	$(MAKE) -C $(@D) $(@F)
 
-# Create links for shared libraries using the `ldconfig' program if possible.
-# Ignore the error if we cannot update /etc/ld.so.cache.
-ifeq (no,$(cross-compiling))
-ifeq (yes,$(build-shared))
-install: install-symbolic-link
-.PHONY: install-symbolic-link
-install-symbolic-link: subdir_install
-	$(symbolic-link-prog) $(symbolic-link-list)
-	rm -f $(symbolic-link-list)
-
-install:
-	-test ! -x $(common-objpfx)elf/ldconfig || LC_ALL=C LANGUAGE=C \
-	  $(common-objpfx)elf/ldconfig $(addprefix -r ,$(install_root)) \
-				       $(slibdir) $(libdir)
-ifneq (no,$(PERL))
-ifeq (/usr,$(prefix))
-ifeq (,$(install_root))
-	CC="$(CC)" $(PERL) scripts/test-installation.pl $(common-objpfx)
-endif
-endif
-endif
-endif
-endif
 
 # Build subdirectory lib objects.
