diff -r -U2 glibc-2.3.3/sysdeps/i386/dl-machine.h glibc-2.3.3/sysdeps/i386/dl-machine.h
--- glibc-2.3.3/sysdeps/i386/dl-machine.h	2003-09-24 14:08:31.000000000 -0700
+++ glibc-2.3.3/sysdeps/i386/dl-machine.h	2003-09-24 14:08:31.000000000 -0700
@@ -155,9 +155,4 @@
 /* GKM FIXME: Fix trampoline to pass bounds so we can do
    without the `__unbounded' qualifier.  */
-static ElfW(Addr) fixup (struct link_map *__unbounded l, ElfW(Word) reloc_offset)
-     __attribute__ ((regparm (2), unused));
-static ElfW(Addr) profile_fixup (struct link_map *l, ElfW(Word) reloc_offset,
-				 ElfW(Addr) retaddr)
-     __attribute__ ((regparm (3), unused));
 # endif
 
