diff -U2 -r glibc-2.2.5/Makefile glibc-2.2.5/Makefile
--- glibc-2.2.5/Makefile	2002-01-08 14:43:50.000000000 -0700
+++ glibc-2.2.5/Makefile	2002-01-08 14:43:50.000000000 -0700
@@ -111,5 +111,5 @@
 	-test ! -x $(common-objpfx)elf/ldconfig || LC_ALL=C LANGUAGE=C \
 	  $(common-objpfx)elf/ldconfig $(addprefix -r ,$(install_root)) \
-				       $(slibdir) $(libdir)
+				       $(slibdir) $(libdir) -N -n
 ifneq (no,$(PERL))
 ifeq (/usr,$(prefix))
diff -U2 -r glibc-2.2.5/sunrpc/Makefile glibc-2.2.5/sunrpc/Makefile
--- glibc-2.2.5/sunrpc/Makefile	2001-11-01 19:07:50.000000000 -0700
+++ glibc-2.2.5/sunrpc/Makefile	2001-11-01 19:07:50.000000000 -0700
@@ -127,5 +127,5 @@
 
 # Tell rpcgen where to find the C preprocessor.
-rpcgen-cmd = $(built-program-cmd) -Y ../scripts
+rpcgen-cmd = $(built-program-cmd)
 
 # Install the rpc data base file.
diff -U2 -r glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c
--- glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c	2001-07-05 21:56:12.000000000 -0700
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c	2001-07-05 21:56:12.000000000 -0700
@@ -39,8 +39,7 @@
 
 strong_alias (__old_sys_nerr, _old_sys_nerr);
-weak_alias (__old_sys_nerr, _old_sys_nerr)
 compat_symbol (libc, __old_sys_nerr, _sys_nerr, GLIBC_2_0);
 compat_symbol (libc, _old_sys_nerr, sys_nerr, GLIBC_2_0);
-weak_alias (__old_sys_errlist, _old_sys_errlist);
+strong_alias (__old_sys_errlist, _old_sys_errlist);
 compat_symbol (libc, __old_sys_errlist, _sys_errlist, GLIBC_2_0);
 compat_symbol (libc, _old_sys_errlist, sys_errlist, GLIBC_2_0);
