diff -r -U2 wine-1.0.1/dlls/gdi32/freetype.c wine-1.0.1/dlls/gdi32/freetype.c
--- wine-1.0.1/dlls/gdi32/freetype.c	2008-10-17 08:45:25.000000000 -0700
+++ wine-1.0.1/dlls/gdi32/freetype.c	2008-10-17 08:45:25.000000000 -0700
@@ -164,5 +164,9 @@
 MAKE_FUNCPTR(FT_Load_Glyph);
 MAKE_FUNCPTR(FT_Matrix_Multiply);
+#ifdef FT_MULFIX_INLINED
+#define pFT_MulFix FT_MULFIX_INLINED
+#else
 MAKE_FUNCPTR(FT_MulFix);
+#endif 
 MAKE_FUNCPTR(FT_New_Face);
 MAKE_FUNCPTR(FT_New_Memory_Face);
@@ -2439,5 +2443,7 @@
     LOAD_FUNCPTR(FT_Load_Glyph)
     LOAD_FUNCPTR(FT_Matrix_Multiply)
+#ifndef FT_MULFIX_INLINED
     LOAD_FUNCPTR(FT_MulFix)
+#endif
     LOAD_FUNCPTR(FT_New_Face)
     LOAD_FUNCPTR(FT_New_Memory_Face)
