diff -r -U2 OpenGTL-0.9.14/CMakeLists.txt OpenGTL-0.9.14/CMakeLists.txt
--- OpenGTL-0.9.14/CMakeLists.txt	2010-06-09 05:57:41.000000000 -0700
+++ OpenGTL-0.9.14/CMakeLists.txt	2010-06-09 05:57:41.000000000 -0700
@@ -45,9 +45,4 @@
 configure_file(config-endian.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-endian.h )
 
-find_package(LLVM REQUIRED)
-if( NOT MSVC AND NOT LLVM_VERSION STREQUAL "2.7" ) # There is no way with MSVC to know the llvm version
-  message(FATAL_ERROR "LLVM 2.7 is required.")
-endif( NOT MSVC AND NOT LLVM_VERSION STREQUAL "2.7" )
-
 if(MSVC)
   add_definitions(/FIiso646.h)
@@ -77,12 +72,4 @@
 endif(OPENGTL_ENABLE_DEBUG_OUTPUT)
 
-if(NOT LLVM_COMPILED_WITHOUT_ASSERTS)
-  if(OVERIDE_LLVM_ASSERT)
-    message(STATUS "llvm was build with asserts, this is not supported by OpenGTL but you have enable the overide")
-  else()
-    message(FATAL_ERROR "llvm was build with asserts, this is not supported by OpenGTL (llvm with asserts is not thread safe).")
-  endif()
-endif(NOT LLVM_COMPILED_WITHOUT_ASSERTS)
-
 if(OPENGTL_FULL_WARNINGS)
   set(OPENGTL_BUILD_DEFINITION "-Wall -Werror")
