diff -r -U2 fontconfig-2.2.1/fontconfig/fcfreetype.h fontconfig-2.2.1/fontconfig/fcfreetype.h
--- fontconfig-2.2.1/fontconfig/fcfreetype.h	2003-03-28 10:08:35.000000000 -0700
+++ fontconfig-2.2.1/fontconfig/fcfreetype.h	2003-03-28 10:08:35.000000000 -0700
@@ -25,5 +25,6 @@
 #ifndef _FCFREETYPE_H_
 #define _FCFREETYPE_H_
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 
 FT_UInt
diff -r -U2 fontconfig-2.2.1/src/fcfreetype.c fontconfig-2.2.1/src/fcfreetype.c
--- fontconfig-2.2.1/src/fcfreetype.c	2003-04-30 08:17:58.000000000 -0700
+++ fontconfig-2.2.1/src/fcfreetype.c	2003-04-30 08:17:58.000000000 -0700
@@ -49,14 +49,15 @@
 #include <string.h>
 #include "fcint.h"
-#include <freetype/freetype.h>
-#include <freetype/internal/ftobjs.h>
-#include <freetype/tttables.h>
-#include <freetype/ftsnames.h>
-#include <freetype/ttnameid.h>
-#include <freetype/t1tables.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_INTERNAL_OBJECTS_H
+#include FT_TRUETYPE_TABLES_H
+#include FT_SFNT_NAMES_H
+#include FT_TRUETYPE_IDS_H
+#include FT_TYPE1_TABLES_H
 
 #if (FREETYPE_MINOR > 1 || (FREETYPE_MINOR == 1 && FREETYPE_PATCH >= 4))
-#include <freetype/ftbdf.h>
-#include <freetype/ftmodule.h>
+#include FT_BDF_H
+// #include FT_CONFIG_MODULES_H
 #define USE_FTBDF
 #define HAS_BDF_PROPERTY(f) ((f) && (f)->driver && \
