diff -r -U2 libxcb-1.0/src/xcb_conn.c libxcb-1.0/src/xcb_conn.c
--- libxcb-1.0/src/xcb_conn.c	2006-11-21 21:18:48.000000000 -0700
+++ libxcb-1.0/src/xcb_conn.c	2006-11-21 21:18:48.000000000 -0700
@@ -63,4 +63,9 @@
 {
     xlib->lock = 0;
+    xlib->sloppy_lock = 0;
+
+    if (getenv("LIBXCB_ALLOW_SLOPPY_LOCK"))
+        xlib->sloppy_lock = 1;
+
     pthread_cond_init(&xlib->cond, 0);
     return 1;
diff -r -U2 libxcb-1.0/src/xcb_xlib.c libxcb-1.0/src/xcb_xlib.c
--- libxcb-1.0/src/xcb_xlib.c	2006-11-02 18:38:00.000000000 -0700
+++ libxcb-1.0/src/xcb_xlib.c	2006-11-02 18:38:00.000000000 -0700
@@ -39,5 +39,6 @@
 {
     _xcb_lock_io(c);
-    assert(!c->xlib.lock);
+    if (!c->xlib.sloppy_lock)
+	    assert(!c->xlib.lock);
     c->xlib.lock = 1;
     c->xlib.thread = pthread_self();
@@ -48,5 +49,6 @@
 {
     _xcb_lock_io(c);
-    assert(c->xlib.lock);
+    if (!c->xlib.sloppy_lock)
+	assert(c->xlib.lock);
     assert(pthread_equal(c->xlib.thread, pthread_self()));
     c->xlib.lock = 0;
diff -r -U2 libxcb-1.0/src/xcbint.h libxcb-1.0/src/xcbint.h
--- libxcb-1.0/src/xcbint.h	2006-11-21 00:06:29.000000000 -0700
+++ libxcb-1.0/src/xcbint.h	2006-11-21 00:06:29.000000000 -0700
@@ -131,4 +131,5 @@
 typedef struct _xcb_xlib {
     int lock;
+    int sloppy_lock;
     pthread_t thread;
     pthread_cond_t cond;
