diff -r -U2 gcc-3.3.2/gcc/Makefile.in gcc-3.3.2/gcc/Makefile.in
--- gcc-3.3.2/gcc/Makefile.in	2003-08-10 16:11:19.000000000 -0700
+++ gcc-3.3.2/gcc/Makefile.in	2003-08-10 16:11:19.000000000 -0700
@@ -2370,5 +2370,4 @@
 	SHELL='$(SHELL)' ;\
 	export TARGET_MACHINE srcdir SHELL ; \
-	$(SHELL) ./fixinc.sh `${PWD_COMMAND}`/include $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS); \
 	rm -f include/syslimits.h; \
 	if [ -f include/limits.h ]; then \
