diff -r -U2 qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/common.pri qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/common.pri
--- qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/common.pri	2012-11-23 03:10:05.000000000 -0700
+++ qt-everywhere-opensource-src-4.8.4/src/3rdparty/webkit/Source/common.pri	2012-11-23 03:10:05.000000000 -0700
@@ -4,10 +4,4 @@
 contains(JAVASCRIPTCORE_JIT,no): DEFINES+=ENABLE_JIT=0
 
-linux-g++ {
-isEmpty($$(SBOX_DPKG_INST_ARCH)):exists(/usr/bin/ld.gold) {
-    message(Using gold linker)
-    QMAKE_LFLAGS+=-fuse-ld=gold
-}
-}
 
 # We use this flag on production branches
