diff -r -U2 shishi-1.0.0/lib/starttls.c shishi-1.0.0/lib/starttls.c
--- shishi-1.0.0/lib/starttls.c	2010-04-28 23:43:49.000000000 -0700
+++ shishi-1.0.0/lib/starttls.c	2010-04-28 23:43:49.000000000 -0700
@@ -126,18 +126,4 @@
     shishi_error_printf (handle, "TLS handshake completed (not resumed)");
 
-  if (have_cas)
-    {
-      ret = gnutls_certificate_verify_peers2 (session, &status);
-      if (ret != 0 || status != 0)
-	{
-	  shishi_error_printf (handle,
-			       "TLS verification of CA failed (%d/%d)", ret,
-			       status);
-	  return SHISHI_RECVFROM_ERROR;
-	}
-
-      /* XXX: We need to verify the CA cert further here. */
-    }
-
   if (session_data_size == 0)
     {
diff -r -U2 shishi-1.0.0/src/starttls.c shishi-1.0.0/src/starttls.c
--- shishi-1.0.0/src/starttls.c	2010-04-28 23:43:49.000000000 -0700
+++ shishi-1.0.0/src/starttls.c	2010-04-28 23:43:49.000000000 -0700
@@ -196,8 +196,4 @@
   gnutls_x509_crt_deinit (cert);
 
-  rc = gnutls_certificate_verify_peers (session);
-  if (rc != GNUTLS_E_SUCCESS)
-    syslog (LOG_ERR, "TLS client certificate verify failed (%d): %s",
-	    rc, gnutls_strerror (rc));
 }
 
