diff -r -U2 console-tools-0.2.3/contrib/codepage.c console-tools-0.2.3/contrib/codepage.c
--- console-tools-0.2.3/contrib/codepage.c	1999-04-08 14:29:41.000000000 -0700
+++ console-tools-0.2.3/contrib/codepage.c	1999-04-08 14:29:41.000000000 -0700
@@ -228,7 +228,5 @@
 	  return 0;
 
-        fprintf(stderr, "\
-Warning: CP format is a hack!\n
-         The files produced may or may not be usable!\n");
+        fprintf(stderr, "\nWarning: CP format is a hack!\n         The files produced may or may not be usable!\n");
 
 	sprintf(outfile, "%d.cp", CPEntryHeader.codepage);
