diff -r -U2 kde-workspace-4.10.4/CMakeLists.txt kde-workspace-4.10.4/CMakeLists.txt
--- kde-workspace-4.10.4/CMakeLists.txt	2013-05-30 12:46:41.000000000 -0700
+++ kde-workspace-4.10.4/CMakeLists.txt	2013-05-30 12:46:41.000000000 -0700
@@ -20,5 +20,5 @@
 include(KDE4Defaults)
 
-find_package(Strigi REQUIRED)
+find_package(Strigi)
 find_package(ZLIB REQUIRED)
 find_package(KActivities 6.0.0 REQUIRED)
diff -r -U2 kde-workspace-4.10.4/kcontrol/kfontinst/CMakeLists.txt kde-workspace-4.10.4/kcontrol/kfontinst/CMakeLists.txt
--- kde-workspace-4.10.4/kcontrol/kfontinst/CMakeLists.txt	2013-05-28 11:37:57.000000000 -0700
+++ kde-workspace-4.10.4/kcontrol/kfontinst/CMakeLists.txt	2013-05-28 11:37:57.000000000 -0700
@@ -40,5 +40,4 @@
     add_subdirectory( dbus )
     add_subdirectory( kcmfontinst ) 
-    add_subdirectory( strigi-analyzer )
     add_subdirectory( apps )
     add_subdirectory( kio ) 
diff -r -U2 kde-workspace-4.10.4/plasma/desktop/applets/CMakeLists.txt kde-workspace-4.10.4/plasma/desktop/applets/CMakeLists.txt
--- kde-workspace-4.10.4/plasma/desktop/applets/CMakeLists.txt	2013-05-28 11:37:56.000000000 -0700
+++ kde-workspace-4.10.4/plasma/desktop/applets/CMakeLists.txt	2013-05-28 11:37:56.000000000 -0700
@@ -1,3 +1,2 @@
-add_subdirectory(kickoff)
 add_subdirectory(showActivityManager)
 add_subdirectory(trash)
