diff -r -U2 texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in
--- texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in	2009-10-14 16:39:50.000000000 -0700
+++ texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in	2009-10-14 16:39:50.000000000 -0700
@@ -574,8 +574,5 @@
 	  rm -f getnonfreefonts-sys && \
 	  echo "creating link 'getnonfreefonts-sys' -> 'getnonfreefonts'" && \
-	  $(LN_S) getnonfreefonts getnonfreefonts-sys && \
-	  rm -f man && \
-	  echo "creating link 'man' -> '$(REL)/texmf/doc/man'" && \
-	  (ln -s $(REL)/texmf/doc/man man || :)
+	  $(LN_S) getnonfreefonts getnonfreefonts-sys
 
 uninstall-hook:
