diff -r -U2 linux-2.6.36.1/fs/proc/base.c linux-2.6.36.1/fs/proc/base.c
--- linux-2.6.36.1/fs/proc/base.c	2010-11-22 12:03:49.000000000 -0700
+++ linux-2.6.36.1/fs/proc/base.c	2010-11-22 12:03:49.000000000 -0700
@@ -1527,5 +1527,5 @@
 		return -ENOMEM;
 
-	pathname = d_path_with_unreachable(path, tmp, PAGE_SIZE);
+	pathname = d_path(path, tmp, PAGE_SIZE);
 	len = PTR_ERR(pathname);
 	if (IS_ERR(pathname))
