diff -U2 -r gnome-pim-1.90.1/gncal/doc/C/Makefile.am gnome-pim-1.90.1/gncal/doc/C/Makefile.am
--- gnome-pim-1.90.1/gncal/doc/C/Makefile.am	2002-06-01 17:24:52.000000000 -0700
+++ gnome-pim-1.90.1/gncal/doc/C/Makefile.am	2002-06-01 17:24:52.000000000 -0700
@@ -13,3 +13,2 @@
 install-data-local: gnomecal.xml
 	mkdir -p $(DESTDIR)$(helpdir)
-	db2html -o $(DESTDIR)$(helpdir) $<
diff -U2 -r gnome-pim-1.90.1/gncal/doc/C/Makefile.in gnome-pim-1.90.1/gncal/doc/C/Makefile.in
--- gnome-pim-1.90.1/gncal/doc/C/Makefile.in	2002-06-01 17:28:54.000000000 -0700
+++ gnome-pim-1.90.1/gncal/doc/C/Makefile.in	2002-06-01 17:28:54.000000000 -0700
@@ -402,5 +402,4 @@
 install-data-local: gnomecal.xml
 	mkdir -p $(DESTDIR)$(helpdir)
-	db2html -o $(DESTDIR)$(helpdir) $<
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
diff -U2 -r gnome-pim-1.90.1/gnomecard/help/C/Makefile.am gnome-pim-1.90.1/gnomecard/help/C/Makefile.am
--- gnome-pim-1.90.1/gnomecard/help/C/Makefile.am	2002-03-07 17:16:39.000000000 -0700
+++ gnome-pim-1.90.1/gnomecard/help/C/Makefile.am	2002-03-07 17:16:39.000000000 -0700
@@ -11,3 +11,2 @@
 install-data-local: gnomecard.xml
 	mkdir -p $(DESTDIR)$(helpdir)
-	db2html -o $(DESTDIR)$(helpdir) $<
diff -U2 -r gnome-pim-1.90.1/gnomecard/help/C/Makefile.in gnome-pim-1.90.1/gnomecard/help/C/Makefile.in
--- gnome-pim-1.90.1/gnomecard/help/C/Makefile.in	2002-06-01 12:36:38.000000000 -0700
+++ gnome-pim-1.90.1/gnomecard/help/C/Makefile.in	2002-06-01 12:36:38.000000000 -0700
@@ -283,5 +283,4 @@
 install-data-local: gnomecard.xml
 	mkdir -p $(DESTDIR)$(helpdir)
-	db2html -o $(DESTDIR)$(helpdir) $<
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
