diff -r -U2 gdm-2.16.1/config/Xsession.in gdm-2.16.1/config/Xsession.in
--- gdm-2.16.1/config/Xsession.in	2006-03-28 21:41:49.000000000 -0700
+++ gdm-2.16.1/config/Xsession.in	2006-03-28 21:41:49.000000000 -0700
@@ -234,4 +234,8 @@
 fi
 
+# launch using dbus-launch if found
+[ -x     /usr/bin/dbus-launch ]  &&
+command="/usr/bin/dbus-launch $command"
+
 echo "$0: Setup done, will execute: $command"
 
