--- gcc-3.4.3/gcc/Makefile.in	2005-05-21 10:35:08.613136000 +0200
+++ gcc-3.4.3-new/gcc/Makefile.in	2005-05-21 11:33:50.124784872 +0200
@@ -286,7 +286,7 @@
 # The GCC to use for compiling libgcc.a and crt*.o.
 # Usually the one we just built.
 # Don't use this as a dependency--use $(GCC_PASSES) or $(GCC_PARTS).
-GCC_FOR_TARGET = $(STAGE_CC_WRAPPER) ./xgcc -B./ -B$(build_tooldir)/bin/ -isystem $(build_tooldir)/include -isystem $(build_tooldir)/sys-include -L$(objdir)/../ld
+GCC_FOR_TARGET = $(STAGE_CC_WRAPPER) ./xgcc -B./ -B$(build_tooldir)/bin/ -isystem $(build_tooldir)/include -isystem $(local_prefix)/include -isystem $(build_tooldir)/sys-include -L$(objdir)/../ld
 
 # This is used instead of ALL_CFLAGS when compiling with GCC_FOR_TARGET.
 # It omits XCFLAGS, and specifies -B./.
@@ -304,7 +304,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo ar; \
     else \
-       t='$(program_transform_name)'; echo ar | sed -e $$t ; \
+      echo ar | sed -e '$(program_transform_name)'; \
     fi; \
   fi`
 AR_FLAGS_FOR_TARGET =
@@ -317,7 +317,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo $(RANLIB); \
     else \
-       t='$(program_transform_name)'; echo ranlib | sed -e $$t ; \
+      echo ranlib | sed -e '$(program_transform_name)' ; \
     fi; \
   fi`
 NM_FOR_TARGET = ` \
@@ -329,7 +329,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo nm; \
     else \
-       t='$(program_transform_name)'; echo nm | sed -e $$t ; \
+      echo nm | sed -e '$(program_transform_name)' ; \
     fi; \
   fi`
 
