diff -r -U2 gcc-20021007/gcc/Makefile.in gcc-20021007/gcc/Makefile.in
--- gcc-20021007/gcc/Makefile.in	2002-10-05 11:55:54.000000000 -0700
+++ gcc-20021007/gcc/Makefile.in	2002-10-05 11:55:54.000000000 -0700
@@ -2984,5 +2984,5 @@
 # output of `cd', but some shells lose on redirection within `()'s
 	(cd `${PWD}`/include ; \
-	 tar -cf - .; exit 0) | (cd $(libsubdir)/include; tar xpf - )
+	 tar -cf - .; exit 0) | (cd $(libsubdir)/include; tar xpf -; exit 0 )
 # /bin/sh on some systems returns the status of the first tar,
 # and that can lose with GNU tar which always writes a full block.
