diff -r -U2 gcc-3.2/gcc/Makefile.in gcc-3.2/gcc/Makefile.in
--- gcc-3.2/gcc/Makefile.in	2002-05-23 10:57:21.000000000 -0700
+++ gcc-3.2/gcc/Makefile.in	2002-05-23 10:57:21.000000000 -0700
@@ -2154,5 +2154,4 @@
 	SHELL='$(SHELL)' ;\
 	export TARGET_MACHINE srcdir SHELL ; \
-	$(SHELL) ./fixinc.sh `pwd`/include $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS); \
 	rm -f include/syslimits.h; \
 	if [ -f include/limits.h ]; then \
