diff -r -U2 gcc-3.3/gcc/Makefile.in gcc-3.3/gcc/Makefile.in
--- gcc-3.3/gcc/Makefile.in	2003-04-22 08:50:48.000000000 -0700
+++ gcc-3.3/gcc/Makefile.in	2003-04-22 08:50:48.000000000 -0700
@@ -2998,5 +2998,5 @@
 # output of `cd', but some shells lose on redirection within `()'s
 	(cd `${PWD}`/include ; \
-	 tar -cf - .; exit 0) | (cd $(DESTDIR)$(libsubdir)/include; tar xpf - )
+	 tar -cf - .; exit 0) | (cd $(DESTDIR)$(libsubdir)/include; tar xpf -; exit 0)
 # /bin/sh on some systems returns the status of the first tar,
 # and that can lose with GNU tar which always writes a full block.
