diff -N -r -U2 uqm-0.7.0/build.vars uqm-0.7.0/build.vars
--- uqm-0.7.0/build.vars	1969-12-31 17:00:00.000000000 -0700
+++ uqm-0.7.0/build.vars	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,60 @@
+# Variables for the build procedure
+# Copyright (c) 2002 Serge van den Boom
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+BUILD_SYSTEM='Linux'
+HOST_SYSTEM='Linux'
+PREPROC_C='gcc -E '
+MKDEP_C='gcc -MM '
+COMPILE_C='gcc -c '
+PREPROC_OBJC=''
+MKDEP_OBJC=''
+COMPILE_OBJC=''
+LINK='gcc '
+MAKE='make'
+ECHON='echo -n'
+SED='sed'
+REZ=''
+WINDRES=''
+uqm_CFLAGS=' -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -O3 -DNDEBUG -DGFXMODULE_SDL -DHAVE_OPENGL  -pthread -D_REENTRANT  -DNETPLAY=NETPLAY_FULL  -DHAVE_JOYSTICK -DHAVE_ZIP=1  -DTHREADLIB_SDL -I"."'
+uqm_LDFLAGS=' -lSDL -lpthread -lSDL_image -lGL -lmikmod -ldl -lm -lvorbisfile -lvorbis -lm -logg -lz '
+uqm_INSTALL_BINDIR='/aux/run/destdir/uqm/usr/bin/'
+uqm_INSTALL_LIBDIR='/aux/run/destdir/uqm/usr/lib/'
+uqm_INSTALL_SHAREDIR='/aux/run/destdir/uqm/usr/share/'
+
+# Exported variables are available from all the Makeinfo files
+# Non-exported files only where build.vars is explicitely included.
+uqm_SOUNDMODULE='mixsdl'
+uqm_OGGVORBIS='vorbisfile'
+uqm_USE_INTERNAL_MIKMOD=''
+uqm_HAVE_GETOPT_LONG='#define HAVE_GETOPT_LONG'
+uqm_HAVE_REGEX='1'
+uqm_GFXMODULE='sdl'
+uqm_HAVE_OPENGL='1'
+uqm_USE_ZIP_IO='1'
+uqm_USE_PLATFORM_ACCEL='0'
+uqm_THREADLIB='SDL'
+uqm_NETPLAY='FULL'
+uqm_USE_WINSOCK=''
+DEBUG='0'
+MACRO_WIN32=''
+MACRO___MINGW32__=''
+export BUILD_SYSTEM HOST_SYSTEM DEBUG
+export MACRO_WIN32 MACRO___MINGW32__
+export uqm_SOUNDMODULE uqm_USE_INTERNAL_MIKMOD uqm_HAVE_GETOPT_LONG
+export uqm_HAVE_REGEX uqm_USE_WINSOCK uqm_GFXMODULE uqm_HAVE_OPENGL
+export uqm_USE_ZIP_IO uqm_USE_PLATFORM_ACCEL uqm_THREADLIB uqm_NETPLAY
+
diff -N -r -U2 uqm-0.7.0/config.state uqm-0.7.0/config.state
--- uqm-0.7.0/config.state	1969-12-31 17:00:00.000000000 -0700
+++ uqm-0.7.0/config.state	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,14 @@
+CHOICE_debug_VALUE='nodebug'
+CHOICE_graphics_VALUE='opengl'
+CHOICE_sound_VALUE='mixsdl'
+CHOICE_mikmod_VALUE='external'
+CHOICE_ovcodec_VALUE='standard'
+CHOICE_netplay_VALUE='full'
+CHOICE_joystick_VALUE='enabled'
+CHOICE_ioformat_VALUE='stdio_zip'
+CHOICE_accel_VALUE='plainc'
+CHOICE_threadlib_VALUE='sdl'
+INPUT_install_prefix_VALUE='/aux/run/destdir/uqm/usr'
+INPUT_install_bindir_VALUE='$prefix/bin'
+INPUT_install_libdir_VALUE='$prefix/lib'
+INPUT_install_sharedir_VALUE='$prefix/share'
diff -N -r -U2 uqm-0.7.0/config_unix.h uqm-0.7.0/config_unix.h
--- uqm-0.7.0/config_unix.h	1969-12-31 17:00:00.000000000 -0700
+++ uqm-0.7.0/config_unix.h	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,63 @@
+/* This file contains some compile-time configuration options for *nix
+ * systems.
+ * config_unix.h is generated from config_unix.h.in by build.sh
+ * When building on MS Windows using build.sh (MinGW, Cygwin),
+ * config_win.h is generated from src/config_win.h.in.
+ * When using MSVC on MS Windows, you'll have to edit src/config_vc6.h
+ * manually if you want anything else than the defaults.
+ */
+
+#ifndef _CONFIG_UNIX_H
+#define _CONFIG_UNIX_H
+
+/* Directory where the UQM game data is located */
+#define CONTENTDIR "/usr/share/uqm/content"
+
+/* Directory where game data will be stored */
+#define USERDIR "~/.uqm/"
+
+/* Directory where config files will be stored */
+#define CONFIGDIR USERDIR
+
+/* Directory where supermelee teams will be stored */
+#define MELEEDIR "${UQM_CONFIG_DIR}/teams/"
+
+/* Directory where save games will be stored */
+#define SAVEDIR "${UQM_CONFIG_DIR}/save/"
+
+/* Defined if words are stored with the most significant byte first */
+#undef WORDS_BIGENDIAN
+
+/* Defined if your system has readdir_r of its own */
+#define HAVE_READDIR_R
+
+/* Defined if your system has setenv of its own */
+#define HAVE_SETENV
+
+/* Defined if your system has strupr of its own */
+#undef HAVE_STRUPR
+
+/* Defined if your system has strcasecmp of its own */
+#define HAVE_STRCASECMP_UQM
+		// Not using "HAVE_STRCASECMP" as that conflicts with SDL.
+
+/* Defined if your system has stricmp of its own */
+#undef HAVE_STRICMP
+
+/* Defined if your system has getopt_long */
+#define HAVE_GETOPT_LONG
+
+/* Defined if your system has iswgraph of its own*/
+#define HAVE_ISWGRAPH
+
+/* Defined if your system has wchar_t of its own */
+#define HAVE_WCHAR_T
+
+/* Defined if your system has wint_t of its own */
+#define HAVE_WINT_T
+
+/* Defined if your system has _Bool of its own */
+#define HAVE__BOOL
+
+#endif  /* _CONFIG_UNIX_H */
+
diff -N -r -U2 uqm-0.7.0/uqm-wrapper uqm-0.7.0/uqm-wrapper
--- uqm-0.7.0/uqm-wrapper	1969-12-31 17:00:00.000000000 -0700
+++ uqm-0.7.0/uqm-wrapper	1969-12-31 17:00:00.000000000 -0700
@@ -0,0 +1,4 @@
+#!/bin/sh
+# Wrapper script for starting The Ur-Quan Masters
+exec "/usr/lib/uqm/uqm" "--contentdir=/usr/share/uqm/content" "$@"
+
