diff -r -U2 isapnptools-1.26/include/isapnp/res-access.h isapnptools-1.26/include/isapnp/res-access.h
--- isapnptools-1.26/include/isapnp/res-access.h	2001-05-08 15:14:21.000000000 -0700
+++ isapnptools-1.26/include/isapnp/res-access.h	2001-05-08 15:14:21.000000000 -0700
@@ -64,6 +64,10 @@
 #define TMP_LEN 16
 
+/*
 extern unsigned char serial_identifier[NUM_CARDS + 1][IDENT_LEN];
 extern unsigned char csumdiff[NUM_CARDS + 1];
+*/
+unsigned char serial_identifier[NUM_CARDS + 1][IDENT_LEN];
+unsigned char csumdiff[NUM_CARDS + 1];
 
 extern int csum;
diff -r -U2 isapnptools-1.26/src/Makefile.in isapnptools-1.26/src/Makefile.in
--- isapnptools-1.26/src/Makefile.in	2001-05-08 15:53:18.000000000 -0700
+++ isapnptools-1.26/src/Makefile.in	2001-05-08 15:53:18.000000000 -0700
@@ -73,5 +73,5 @@
 EXTRA_DIST = getopt.h sfnlfn.dat
 
-lib_LIBRARIES = libisapnp.a
+lib_LIBRARIES = libisapnp.so.1
 
 libisapnp_a_SOURCES =  	callbacks.c 	iopl.c 	pnp-access.c 	res-access.c 	realtime.c 	cardinfo.c 	pnp-select.c 	resource.c 	mysnprtf.c 	release.c
@@ -82,8 +82,8 @@
 sbin_PROGRAMS = pnpdump isapnp
 isapnp_SOURCES = isapnp.c isapnp_main.l
-isapnp_LDADD = @LIBOBJS@ libisapnp.a
+isapnp_LDADD = @LIBOBJS@ libisapnp.so.1
 
 pnpdump_SOURCES = pnpdump.c pnpdump_main.c
-pnpdump_LDADD = @LIBOBJS@ libisapnp.a
+pnpdump_LDADD = @LIBOBJS@ libisapnp.so.1
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../include/config.h
@@ -104,8 +104,8 @@
 
 pnpdump_OBJECTS =  pnpdump.o pnpdump_main.o
-pnpdump_DEPENDENCIES =  @LIBOBJS@ libisapnp.a
+pnpdump_DEPENDENCIES =  @LIBOBJS@ libisapnp.so.1
 pnpdump_LDFLAGS = 
 isapnp_OBJECTS =  isapnp.o isapnp_main.o
-isapnp_DEPENDENCIES =  @LIBOBJS@ libisapnp.a
+isapnp_DEPENDENCIES =  @LIBOBJS@ libisapnp.so.1
 isapnp_LDFLAGS = 
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
@@ -114,5 +114,5 @@
 COMPILE = $(CC) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(LDFLAGS) -L. -o $@
 DIST_COMMON =  Makefile.am Makefile.in getopt.c getopt1.c isapnp_main.c
 
@@ -155,10 +155,10 @@
 	done
 	@$(POST_INSTALL)
-	@list='$(lib_LIBRARIES)'; for p in $$list; do \
-	  if test -f $$p; then \
-	    echo " $(RANLIB) $(DESTDIR)$(libdir)/$$p"; \
-	    $(RANLIB) $(DESTDIR)$(libdir)/$$p; \
-	  else :; fi; \
-	done
+#	@list='$(lib_LIBRARIES)'; for p in $$list; do \
+#	  if test -f $$p; then \
+#	    echo " $(RANLIB) $(DESTDIR)$(libdir)/$$p"; \
+#	    $(RANLIB) $(DESTDIR)$(libdir)/$$p; \
+#	  else :; fi; \
+#	done
 
 uninstall-libLIBRARIES:
@@ -187,8 +187,7 @@
 maintainer-clean-compile:
 
-libisapnp.a: $(libisapnp_a_OBJECTS) $(libisapnp_a_DEPENDENCIES)
-	-rm -f libisapnp.a
-	$(AR) cru libisapnp.a $(libisapnp_a_OBJECTS) $(libisapnp_a_LIBADD)
-	$(RANLIB) libisapnp.a
+libisapnp.so.1: $(libisapnp_a_OBJECTS) $(libisapnp_a_DEPENDENCIES)
+	-rm -f libisapnp.so.1
+	$(LINK) -shared -fPIC -Wl,-soname,libisapnp.so.1 $(libisapnp_a_OBJECTS)
 
 mostlyclean-sbinPROGRAMS:
diff -r -U2 isapnptools-1.26/src/isapnp_main.l isapnptools-1.26/src/isapnp_main.l
--- isapnptools-1.26/src/isapnp_main.l	2001-05-08 15:14:21.000000000 -0700
+++ isapnptools-1.26/src/isapnp_main.l	2001-05-08 15:14:21.000000000 -0700
@@ -364,5 +364,5 @@
 }
 
-static unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
+unsigned char serial_identifier[NUM_CARDS+1][IDENT_LEN];
 static char *boardid[NUM_CARDS+1];
 static unsigned long serno[NUM_CARDS+1];
