diff -Naur findutils-4.1/find/defs.h findutils-4.1/find/defs.h
--- findutils-4.1/find/defs.h	Wed Nov  2 15:59:15 1994
+++ findutils-4.1/find/defs.h	Wed Nov  2 15:59:15 1994
@@ -301,7 +301,7 @@
 boolean mark_stat P_((struct predicate *tree));
 
 /* util.c */
-char *basename P_((char *fname));
+char *basename2 P_((char *fname));
 struct predicate *get_new_pred P_((void));
 struct predicate *get_new_pred_chk_op P_((void));
 struct predicate *insert_primary P_((boolean (*pred_func )()));
diff -Naur findutils-4.1/find/fstype.c findutils-4.1/find/fstype.c
--- findutils-4.1/find/fstype.c	Thu Nov  3 11:33:48 1994
+++ findutils-4.1/find/fstype.c	Thu Nov  3 11:33:48 1994
@@ -30,9 +30,6 @@
 extern int errno;
 #endif
 
-char *strdup ();
-char *strstr ();
-
 static char *filesystem_type_uncached P_((char *path, char *relpath, struct stat *statp));
 static int xatoi P_((char *cp));
 
diff -Naur findutils-4.1/find/parser.c findutils-4.1/find/parser.c
--- findutils-4.1/find/parser.c	Wed Nov  2 15:59:19 1994
+++ findutils-4.1/find/parser.c	Wed Nov  2 15:59:19 1994
@@ -52,7 +52,6 @@
 #define lstat stat
 #endif
 
-char *strstr ();
 int lstat ();
 int stat ();
 #ifndef atol /* for Linux */
diff -Naur findutils-4.1/find/pred.c findutils-4.1/find/pred.c
--- findutils-4.1/find/pred.c	Wed Nov  2 15:59:23 1994
+++ findutils-4.1/find/pred.c	Wed Nov  2 15:59:23 1994
@@ -15,6 +15,8 @@
    along with this program; if not, write to the Free Software
    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.  */
 
+#define _GNU_SOURCE
+
 #include <config.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -Naur findutils-4.1/find/util.c findutils-4.1/find/util.c
--- findutils-4.1/find/util.c	Thu Oct 20 00:49:47 1994
+++ findutils-4.1/find/util.c	Thu Oct 20 00:49:47 1994
@@ -25,7 +25,7 @@
    compressed into one slash. */
 
 char *
-basename (fname)
+basename2 (fname)
      char *fname;
 {
   char *p;
diff -Naur findutils-4.1/lib/nextelem.c findutils-4.1/lib/nextelem.c
--- findutils-4.1/lib/nextelem.c	Tue Sep 27 09:02:45 1994
+++ findutils-4.1/lib/nextelem.c	Tue Sep 27 09:02:45 1994
@@ -32,7 +32,6 @@
 #endif
 #endif
 
-char *strdup ();
 void free ();
 
 /* Return the next element of a colon-separated path.
diff -Naur findutils-4.1/xargs/xargs.c findutils-4.1/xargs/xargs.c
--- findutils-4.1/xargs/xargs.c	Fri Oct  7 13:21:39 1994
+++ findutils-4.1/xargs/xargs.c	Fri Oct  7 13:21:39 1994
@@ -60,9 +60,6 @@
 #define memcpy(dest, source, count) (bcopy((source), (dest), (count)))
 #endif
 
-char *strstr ();
-char *strdup ();
-
 #ifndef _POSIX_SOURCE
 #include <sys/param.h>
 #endif
