diff -r -U2 glhack-1.2/win/gl/gl_image.c glhack-1.2/win/gl/gl_image.c
--- glhack-1.2/win/gl/gl_image.c	2002-12-15 04:21:13.000000000 -0700
+++ glhack-1.2/win/gl/gl_image.c	2002-12-15 04:21:13.000000000 -0700
@@ -487,5 +487,5 @@
    * (this is the normal method of doing things with libpng).
    */
-  if (setjmp(png_ptr->jmpbuf))
+  if (setjmp(png_jmpbuf(png_ptr)))
   {
     sdlgl_warning("Problem within LibPNG (unknown)\n");
