diff -r -U2 mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp
--- mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-19 13:55:52.000000000 -0700
+++ mozilla-1.9.2/toolkit/system/gnome/nsAlertsIconListener.cpp	2011-03-19 13:55:52.000000000 -0700
@@ -205,5 +205,5 @@
   NotifyNotification* notify = notify_notification_new(mAlertTitle.get(),
                                                        mAlertText.get(),
-                                                       NULL, NULL);
+                                                       NULL);
   if (!notify)
     return NS_ERROR_OUT_OF_MEMORY;
