--- vobsub.c.orig	2006-05-28 23:48:01.000000000 -0500
+++ vobsub.c	2006-04-24 02:20:33.000000000 -0500
@@ -227,7 +227,7 @@
 /**********************************************************************/
 
 static ssize_t
-getline(char **lineptr, size_t *n, rar_stream_t *stream)
+vobsub_getline(char **lineptr, size_t *n, rar_stream_t *stream)
 {
     size_t res = 0;
     int c;
@@ -949,7 +946,7 @@
 	size_t line_reserve = 0;
 	char *line = NULL;
     do {
-	line_size = getline(&line, &line_reserve, fd);
+	line_size = vobsub_getline(&line, &line_reserve, fd);
 	if (line_size < 0) {
 	    break;
 	}

