diff -ur glibc-2.3.5/Makefile glibc-2.3.5-crossfix/Makefile
--- glibc-2.3.5/Makefile	2005-04-06 04:16:14.000000000 +0200
+++ glibc-2.3.5-crossfix/Makefile	2005-06-02 22:09:39.319115000 +0200
@@ -68,7 +68,7 @@
 		   $(addprefix install-, no-libc.a bin lib data headers others)
 
 headers := limits.h values.h features.h gnu-versions.h bits/libc-lock.h \
-	   bits/xopen_lim.h gnu/libc-version.h
+	   bits/xopen_lim.h gnu/libc-version.h gnu/stubs.h bits/syscall.h
 
 echo-headers: subdir_echo-headers
 
@@ -77,7 +77,7 @@
 vpath %.h $(subdir-dirs)
 
 # What to install.
-install-others = $(inst_includedir)/gnu/stubs.h
+install-headers = $(inst_includedir)/gnu/stubs.h
 install-bin-script =
 
 ifeq (yes,$(build-shared))
diff -ur glibc-2.3.5/sysdeps/unix/sysv/linux/Makefile glibc-2.3.5-crossfix/sysdeps/unix/sysv/linux/Makefile
--- glibc-2.3.5/sysdeps/unix/sysv/linux/Makefile	2004-10-05 01:29:06.000000000 +0200
+++ glibc-2.3.5-crossfix/sysdeps/unix/sysv/linux/Makefile	2005-06-02 21:34:19.161427960 +0200
@@ -26,7 +26,7 @@
 		  sys/ultrasound.h sys/raw.h sys/personality.h sys/epoll.h \
 		  bits/a.out.h
 
-install-others += $(inst_includedir)/bits/syscall.h
+install-headers += $(inst_includedir)/bits/syscall.h
 
 # Generate the list of SYS_* macros for the system calls (__NR_* macros).
 # For bi-arch platforms, the CPU/Makefile defines {32,64}bit-predefine and
