diff -r -U2 bochs-2.5.1/configure bochs-2.5.1/configure
--- bochs-2.5.1/configure	2012-01-06 11:51:12.000000000 -0700
+++ bochs-2.5.1/configure	2012-01-06 11:51:12.000000000 -0700
@@ -22006,4 +22006,8 @@
 			KERNEL_MODULE_SUFFIX="ko"
 		;;
+		3.*)
+	  		PCIDEV_MODULE_MAKE_ALL="all-kernel26"
+			KERNEL_MODULE_SUFFIX="ko"
+		;;
 		*)
                         as_fn_error "Linux kernel 2.4 or 2.6 is required for PCI host device mapping" "$LINENO" 5
