diff -r -U2 samba-3.4.0/source3/Makefile.in samba-3.4.0/source3/Makefile.in
--- samba-3.4.0/source3/Makefile.in	2009-07-03 04:21:14.000000000 -0700
+++ samba-3.4.0/source3/Makefile.in	2009-07-03 04:21:14.000000000 -0700
@@ -1471,5 +1471,5 @@
 		-lkeyutils $(LIBS) $(LIBSMBCLIENT_OBJ1) $(KRB5LIBS) \
 		$(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBWBCLIENT_LIBS) \
-		$(LIBTDB_LIBS) $(NSCD_LIBS)
+		$(LIBTDB_LIBS) $(NSCD_LIBS) $(ZLIB_LIBS)
 
 bin/testparm@EXEEXT@: $(BINARY_PREREQS) $(TESTPARM_OBJ) @BUILD_POPT@ @LIBTALLOC_TARGET@ @LIBTDB_TARGET@
