diff -r -U2 gcc-4.0.4/Makefile.in gcc-4.0.4/Makefile.in
--- gcc-4.0.4/Makefile.in	2006-04-04 14:04:37.000000000 -0700
+++ gcc-4.0.4/Makefile.in	2006-04-04 14:04:37.000000000 -0700
@@ -669,5 +669,4 @@
     maybe-configure-findutils \
     maybe-configure-find \
-    maybe-configure-fixincludes \
     maybe-configure-flex \
     maybe-configure-gas \
@@ -749,6 +748,5 @@
     maybe-all-build-flex \
     maybe-all-build-m4 \
-    maybe-all-build-texinfo \
-    maybe-all-build-fixincludes
+    maybe-all-build-texinfo
 .PHONY: all-host
 all-host:  \
@@ -771,5 +769,4 @@
     maybe-all-findutils \
     maybe-all-find \
-    maybe-all-fixincludes \
     maybe-all-flex \
     maybe-all-gas \
@@ -868,5 +865,4 @@
     maybe-info-findutils \
     maybe-info-find \
-    maybe-info-fixincludes \
     maybe-info-flex \
     maybe-info-gas \
@@ -961,5 +957,4 @@
     maybe-dvi-findutils \
     maybe-dvi-find \
-    maybe-dvi-fixincludes \
     maybe-dvi-flex \
     maybe-dvi-gas \
@@ -1054,5 +1049,4 @@
     maybe-html-findutils \
     maybe-html-find \
-    maybe-html-fixincludes \
     maybe-html-flex \
     maybe-html-gas \
@@ -1147,5 +1141,4 @@
     maybe-TAGS-findutils \
     maybe-TAGS-find \
-    maybe-TAGS-fixincludes \
     maybe-TAGS-flex \
     maybe-TAGS-gas \
@@ -1240,5 +1233,4 @@
     maybe-install-info-findutils \
     maybe-install-info-find \
-    maybe-install-info-fixincludes \
     maybe-install-info-flex \
     maybe-install-info-gas \
@@ -1333,5 +1325,4 @@
     maybe-installcheck-findutils \
     maybe-installcheck-find \
-    maybe-installcheck-fixincludes \
     maybe-installcheck-flex \
     maybe-installcheck-gas \
@@ -1426,5 +1417,4 @@
     maybe-mostlyclean-findutils \
     maybe-mostlyclean-find \
-    maybe-mostlyclean-fixincludes \
     maybe-mostlyclean-flex \
     maybe-mostlyclean-gas \
@@ -1519,5 +1509,4 @@
     maybe-clean-findutils \
     maybe-clean-find \
-    maybe-clean-fixincludes \
     maybe-clean-flex \
     maybe-clean-gas \
@@ -1612,5 +1601,4 @@
     maybe-distclean-findutils \
     maybe-distclean-find \
-    maybe-distclean-fixincludes \
     maybe-distclean-flex \
     maybe-distclean-gas \
@@ -1705,5 +1693,4 @@
     maybe-maintainer-clean-findutils \
     maybe-maintainer-clean-find \
-    maybe-maintainer-clean-fixincludes \
     maybe-maintainer-clean-flex \
     maybe-maintainer-clean-gas \
@@ -1857,5 +1844,4 @@
     maybe-check-findutils \
     maybe-check-find \
-    maybe-check-fixincludes \
     maybe-check-flex \
     maybe-check-gas \
@@ -1970,5 +1956,4 @@
     maybe-install-findutils \
     maybe-install-find \
-    maybe-install-fixincludes \
     maybe-install-flex \
     maybe-install-gas \
@@ -2038,5 +2023,4 @@
     maybe-install-findutils \
     maybe-install-find \
-    maybe-install-fixincludes \
     maybe-install-flex \
     maybe-install-gas \
@@ -35116,13 +35100,5 @@
 all-stagefeedback-gcc: maybe-all-build-libiberty
 all-prebootstrap: maybe-all-build-libiberty
-all-gcc: maybe-all-build-fixincludes
 
-all-stage1-gcc: maybe-all-build-fixincludes
-all-stage2-gcc: maybe-all-build-fixincludes
-all-stage3-gcc: maybe-all-build-fixincludes
-all-stage4-gcc: maybe-all-build-fixincludes
-all-stageprofile-gcc: maybe-all-build-fixincludes
-all-stagefeedback-gcc: maybe-all-build-fixincludes
-all-prebootstrap: maybe-all-build-fixincludes
 all-gcc: maybe-all-binutils
 
@@ -35173,6 +35149,4 @@
 all-stageprofile-gcc: maybe-all-stageprofile-libiberty
 all-stagefeedback-gcc: maybe-all-stagefeedback-libiberty
-all-gcc: maybe-all-fixincludes
-install-gcc: maybe-install-fixincludes
 configure-libcpp: configure-libiberty
 
