diff -r -U2 xorg-server-1.4/config/dbus.c xorg-server-1.4/config/dbus.c
--- xorg-server-1.4/config/dbus.c	2007-08-23 12:04:52.000000000 -0700
+++ xorg-server-1.4/config/dbus.c	2007-08-23 12:04:52.000000000 -0700
@@ -354,6 +354,7 @@
     dbus_error_init(&error);
 
-    if (!dbus_bus_request_name(info->connection, info->busname,
-                               0, &error)) {
+    dbus_bus_request_name(info->connection, info->busname, 0, &error);
+
+    if (dbus_error_is_set(&error)) {
         ErrorF("[config/dbus] couldn't take over org.x.config: %s (%s)\n",
                error.name, error.message);
