diff -r -U2 Mesa-7.8.1/src/mesa/main/imports.c Mesa-7.8.1/src/mesa/main/imports.c
--- Mesa-7.8.1/src/mesa/main/imports.c	2010-04-04 16:46:26.000000000 -0700
+++ Mesa-7.8.1/src/mesa/main/imports.c	2010-04-04 16:46:26.000000000 -0700
@@ -800,15 +800,5 @@
 _mesa_strtof( const char *s, char **end )
 {
-#ifdef _GNU_SOURCE
-   static locale_t loc = NULL;
-   if (!loc) {
-      loc = newlocale(LC_CTYPE_MASK, "C", NULL);
-   }
-   return strtof_l(s, end, loc);
-#elif defined(_ISOC99_SOURCE) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 600)
-   return strtof(s, end);
-#else
    return (float)strtod(s, end);
-#endif
 }
 
