diff -ur Mesa-7.10.2-r200fix//src/mesa/drivers/dri/r200/r200_cmdbuf.c Mesa-7.10.2/src/mesa/drivers/dri/r200/r200_cmdbuf.c
--- Mesa-7.10.2-r200fix//src/mesa/drivers/dri/r200/r200_cmdbuf.c	2011-06-17 13:58:37.435110211 +0200
+++ Mesa-7.10.2/src/mesa/drivers/dri/r200/r200_cmdbuf.c	2010-12-14 22:43:15.000000000 +0100
@@ -45,11 +45,12 @@
 /* The state atoms will be emitted in the order they appear in the atom list,
  * so this step is important.
  */
-inline void insert_at_tail_if(struct radeon_state_atom *atom_list, struct radeon_state_atom* atom)
-{
-      if (atom->check)
-	 insert_at_tail((atom_list), atom);
-}
+#define insert_at_tail_if(atom_list, atom) \
+   do { \
+      struct radeon_state_atom* __atom = (atom); \
+      if (__atom->check) \
+	 insert_at_tail((atom_list), __atom); \
+   } while(0)
 
 void r200SetUpAtomList( r200ContextPtr rmesa )
 {
