--- ./src/mesa/drivers/dri/savage/savagetex.c.drop-static-inline	2005-10-10 15:15:48.000000000 -0400
+++ ./src/mesa/drivers/dri/savage/savagetex.c	2006-04-02 00:18:29.000000000 -0500
@@ -101,7 +101,7 @@
  * \param w   width in bytes
  */
 #define SUBTILE_FUNC(w,h)					\
-static __inline GLubyte *savageUploadSubtile_##w##x##h		\
+static GLubyte *savageUploadSubtile_##w##x##h		\
 (GLubyte *dest, GLubyte *src, GLuint srcStride)			\
 {								\
     GLuint y;							\
--- ./src/mesa/drivers/dri/savage/savagetris.c.drop-static-inline	2005-10-10 15:15:48.000000000 -0400
+++ ./src/mesa/drivers/dri/savage/savagetris.c	2006-04-02 00:19:03.000000000 -0500
@@ -215,7 +215,7 @@
    tmp.f[vertex_size-1] *= rhw;				\
 } while (0)
 
-static void __inline__ savage_ptex_tri (savageContextPtr imesa,
+static void savage_ptex_tri (savageContextPtr imesa,
 					savageVertexPtr v0,
 					savageVertexPtr v1,
 					savageVertexPtr v2) {
@@ -229,7 +229,7 @@
    PTEX_VERTEX (j, tmp, vertsize, 0, v2); EMIT_VERT (j, vb, vertsize, 0, &amp;tmp);
 }
 
-static __inline__ void savage_ptex_line (savageContextPtr imesa,
+static void savage_ptex_line (savageContextPtr imesa,
 					 savageVertexPtr v0,
 					 savageVertexPtr v1 ) {
    GLuint vertsize = imesa-&gt;HwVertexSize;
@@ -275,7 +275,7 @@
    EMIT_VERT (j, vb, vertsize, 2, &amp;tmp1);
 } 
 
-static __inline__ void savage_ptex_point (savageContextPtr imesa,
+static void savage_ptex_point (savageContextPtr imesa,
 					  savageVertexPtr v0) {
    GLuint vertsize = imesa-&gt;HwVertexSize;
    u_int32_t *vb = savageAllocVtxBuf (imesa, 6*vertsize);


