diff -r -U2 gcc-4.6.2/gcc/Makefile.in gcc-4.6.2/gcc/Makefile.in
--- gcc-4.6.2/gcc/Makefile.in	2011-08-20 00:51:09.000000000 -0700
+++ gcc-4.6.2/gcc/Makefile.in	2011-08-20 00:51:09.000000000 -0700
@@ -4164,11 +4164,4 @@
 	    $(mkinstalldirs) $${fix_dir}; \
 	    chmod a+rx $${fix_dir} || true; \
-	    (TARGET_MACHINE='$(target)'; srcdir=`cd $(srcdir); ${PWD_COMMAND}`; \
-	      SHELL='$(SHELL)'; MACRO_LIST=`${PWD_COMMAND}`/macro_list ; \
-	      gcc_dir=`${PWD_COMMAND}` ; \
-	      export TARGET_MACHINE srcdir SHELL MACRO_LIST && \
-	      cd $(build_objdir)/fixincludes && \
-	      $(SHELL) ./fixinc.sh "$${gcc_dir}/$${fix_dir}" \
-	        $(SYSTEM_HEADER_DIR) $(OTHER_FIXINCLUDES_DIRS) ); \
 	    rm -f $${fix_dir}/syslimits.h; \
 	    if [ -f $${fix_dir}/limits.h ]; then \
diff -r -U2 gcc-4.6.2/gcc/common.opt gcc-4.6.2/gcc/common.opt
--- gcc-4.6.2/gcc/common.opt	2011-04-11 08:26:47.000000000 -0700
+++ gcc-4.6.2/gcc/common.opt	2011-04-11 08:26:47.000000000 -0700
@@ -1780,5 +1780,5 @@
 
 fstack-protector
-Common Report Var(flag_stack_protect, 1)
+Common Report Var(flag_stack_protect, 1) Init(1)
 Use propolice as a stack protection method
 
