diff -r -U2 cdrdao-1.2.2/scsilib/include/schily.h cdrdao-1.2.2/scsilib/include/schily.h
--- cdrdao-1.2.2/scsilib/include/schily.h	2004-03-04 17:30:40.000000000 -0700
+++ cdrdao-1.2.2/scsilib/include/schily.h	2004-03-04 17:30:40.000000000 -0700
@@ -109,5 +109,5 @@
 extern	int	fexecv __PR((const char *, FILE *, FILE *, FILE *, int,
 							char **));
-extern	int	fexecve __PR((const char *, FILE *, FILE *, FILE *,
+extern	int	f_execve __PR((const char *, FILE *, FILE *, FILE *,
 					char * const *, char * const *));
 extern	int	fspawnv __PR((FILE *, FILE *, FILE *, int, char * const *));
@@ -188,5 +188,5 @@
 extern	int	findline __PR((const char *, char, const char *,
 							int, char **, int));
-extern	int	getline __PR((char *, int));
+extern	int	get_line __PR((char *, int));
 extern	int	getstr __PR((char *, int));
 extern	int	breakline __PR((char *, char, char **, int));
diff -r -U2 cdrdao-1.2.2/scsilib/libscg/scsitransp.c cdrdao-1.2.2/scsilib/libscg/scsitransp.c
--- cdrdao-1.2.2/scsilib/libscg/scsitransp.c	2004-06-17 13:20:27.000000000 -0700
+++ cdrdao-1.2.2/scsilib/libscg/scsitransp.c	2004-06-17 13:20:27.000000000 -0700
@@ -324,5 +324,5 @@
 	js_printf("%s", msg);
 	flush();
-	if (getline(okbuf, sizeof (okbuf)) == EOF)
+	if (get_line(okbuf, sizeof (okbuf)) == EOF)
 		exit(EX_BAD);
 	if (streql(okbuf, "y") || streql(okbuf, "yes") ||
diff -r -U2 cdrdao-1.2.2/scsilib/libschily/fexec.c cdrdao-1.2.2/scsilib/libschily/fexec.c
--- cdrdao-1.2.2/scsilib/libschily/fexec.c	2004-06-06 04:50:24.000000000 -0700
+++ cdrdao-1.2.2/scsilib/libschily/fexec.c	2004-06-06 04:50:24.000000000 -0700
@@ -160,5 +160,5 @@
 	va_end(args);
 
-	ret = fexecve(name, in, out, err, av, env);
+	ret = f_execve(name, in, out, err, av, env);
 	if (av != xav)
 		free(av);
@@ -174,9 +174,9 @@
 {
 	av[ac] = NULL;			/*  force list to be null terminated */
-	return (fexecve(name, in, out, err, av, environ));
+	return (f_execve(name, in, out, err, av, environ));
 }
 
 EXPORT int
-fexecve(name, in, out, err, av, env)
+f_execve(name, in, out, err, av, env)
 	const char *name;
 	FILE *in, *out, *err;
diff -r -U2 cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c
--- cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c	2004-08-08 04:02:05.000000000 -0700
+++ cdrdao-1.2.2/scsilib/libschily/stdio/fgetline.c	2004-08-08 04:02:05.000000000 -0700
@@ -65,5 +65,5 @@
 
 EXPORT int
-getline(buf, len)
+get_line(buf, len)
 	char	*buf;
 	int	len;
