diff -r -U2 gwenhywfar-4.0.2/src/sio/syncio_tls.c gwenhywfar-4.0.2/src/sio/syncio_tls.c
--- gwenhywfar-4.0.2/src/sio/syncio_tls.c	2010-08-28 06:08:14.000000000 -0700
+++ gwenhywfar-4.0.2/src/sio/syncio_tls.c	2010-08-28 06:08:14.000000000 -0700
@@ -436,5 +436,4 @@
   gnutls_transport_set_push_function(xio->session, GWEN_SyncIo_Tls_Push);
   gnutls_transport_set_pull_function(xio->session, GWEN_SyncIo_Tls_Pull);
-  gnutls_transport_set_lowat(xio->session, 0);
 
   xio->prepared=1;
