diff -r -U2 qca-tls-1.0/qca-tls.cpp qca-tls-1.0/qca-tls.cpp
--- qca-tls-1.0/qca-tls.cpp	2003-12-10 09:53:57.000000000 -0700
+++ qca-tls-1.0/qca-tls.cpp	2003-12-10 09:53:57.000000000 -0700
@@ -455,5 +455,5 @@
 				// try this other public function, for whatever reason
 				p = (void *)in;
-				r = d2i_RSA_PUBKEY(NULL, (unsigned char **)&p, len);
+				r = d2i_RSA_PUBKEY(NULL, (const unsigned char **)&p, len);
 			}
 			if(r) {
@@ -800,5 +800,5 @@
 	{
 		unsigned char *p = (unsigned char *)in;
-		X509 *t = d2i_X509(NULL, &p, len);
+		X509 *t = d2i_X509(NULL, (const unsigned char **)&p, len);
 		if(!t)
 			return false;
