diff -r -U2 kdesdk-4.10.4/CMakeLists.txt kdesdk-4.10.4/CMakeLists.txt
--- kdesdk-4.10.4/CMakeLists.txt	2013-05-30 15:15:44.000000000 -0700
+++ kdesdk-4.10.4/CMakeLists.txt	2013-05-30 15:15:44.000000000 -0700
@@ -43,5 +43,4 @@
 kdesdk_optional_add_subdirectory(kde-dev-utils)
 kdesdk_optional_add_subdirectory(kdesdk-kioslaves)
-kdesdk_optional_add_subdirectory(kdesdk-strigi-analyzers)
 kdesdk_optional_add_subdirectory(kdesdk-thumbnailers)
 kdesdk_optional_add_subdirectory(kompare)
