diff -r -U2 sysvinit-2.88dsf/src/bootlogd.c sysvinit-2.88dsf/src/bootlogd.c
--- sysvinit-2.88dsf/src/bootlogd.c	2010-03-23 07:37:01.000000000 -0700
+++ sysvinit-2.88dsf/src/bootlogd.c	2010-03-23 07:37:01.000000000 -0700
@@ -368,7 +368,4 @@
 		}
 		switch (*ptr) {
-			case 27: /* ESC */
-				strcpy(tmp, "^[");
-				break;
 			case '\r':
 				line.pos = 0;
@@ -386,4 +383,5 @@
 					line.pos = sizeof(line.buf) - 1;
 				break;
+			case  27:
 			case  32 ... 127:
 			case 161 ... 255:
