diff -r -U2 zlib-1.2.5/zlib.h zlib-1.2.5/zlib.h
--- zlib-1.2.5/zlib.h	2010-04-19 21:12:48.000000000 -0700
+++ zlib-1.2.5/zlib.h	2010-04-19 21:12:48.000000000 -0700
@@ -1581,9 +1581,9 @@
 #  ifdef _LARGEFILE64_SOURCE
      ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
-     ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
-     ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
-     ZEXTERN z_off_t ZEXPORT gzoffset64 OF((gzFile));
-     ZEXTERN uLong ZEXPORT adler32_combine64 OF((uLong, uLong, z_off_t));
-     ZEXTERN uLong ZEXPORT crc32_combine64 OF((uLong, uLong, z_off_t));
+     ZEXTERN z_off64_t ZEXPORT gzseek64 OF((gzFile, z_off64_t, int));
+     ZEXTERN z_off64_t ZEXPORT gztell64 OF((gzFile));
+     ZEXTERN z_off64_t ZEXPORT gzoffset64 OF((gzFile));
+     ZEXTERN uLong ZEXPORT adler32_combine64 OF((uLong, uLong, z_off64_t));
+     ZEXTERN uLong ZEXPORT crc32_combine64 OF((uLong, uLong, z_off64_t));
 #  endif
 #else
