diff -r -U2 coreutils-8.21/src/cat.c coreutils-8.21/src/cat.c
--- coreutils-8.21/src/cat.c	2013-01-30 17:45:59.000000000 -0700
+++ coreutils-8.21/src/cat.c	2013-01-30 17:45:59.000000000 -0700
@@ -95,4 +95,5 @@
   -A, --show-all           equivalent to -vET\n\
   -b, --number-nonblank    number nonempty output lines, overrides -n\n\
+  -d, --direct             open files with flag O_DIRECT \n\
   -e                       equivalent to -vE\n\
   -E, --show-ends          display $ at end of each line\n\
@@ -548,4 +549,5 @@
   {
     {"number-nonblank", no_argument, NULL, 'b'},
+    {"direct", no_argument, NULL, 'd'},
     {"number", no_argument, NULL, 'n'},
     {"squeeze-blank", no_argument, NULL, 's'},
@@ -573,5 +575,5 @@
   /* Parse command line options.  */
 
-  while ((c = getopt_long (argc, argv, "benstuvAET", long_options, NULL))
+  while ((c = getopt_long (argc, argv, "bdenstuvAET", long_options, NULL))
          != -1)
     {
@@ -583,4 +585,8 @@
           break;
 
+        case 'd':
+         file_open_mode |= O_DIRECT;
+         break;
+
         case 'e':
           show_ends = true;
