diff -r -U2 prboom-plus-2.5.0.3/config.h prboom-plus-2.5.0.3/config.h
--- prboom-plus-2.5.0.3/config.h	2009-06-30 02:14:06.000000000 -0700
+++ prboom-plus-2.5.0.3/config.h	2009-06-30 02:14:06.000000000 -0700
@@ -10,5 +10,5 @@
 
 /* Define to be the path where Doom WADs are stored */
-#define DOOMWADDIR "/usr/local/share/games/doom"
+#define DOOMWADDIR "/usr/share/games/doom"
 
 /* Define if you are building with OpenGL support */
diff -r -U2 prboom-plus-2.5.0.3/src/Makefile.am prboom-plus-2.5.0.3/src/Makefile.am
--- prboom-plus-2.5.0.3/src/Makefile.am	2009-06-30 02:02:22.000000000 -0700
+++ prboom-plus-2.5.0.3/src/Makefile.am	2009-06-30 02:02:22.000000000 -0700
@@ -9,5 +9,5 @@
 SUBDIRS = SDL POSIX MAC PCSOUND TEXTSCREEN
 
-gamesdir=$(prefix)/games
+gamesdir=$(prefix)/bin
 
 if BUILD_SERVER
diff -r -U2 prboom-plus-2.5.0.3/src/Makefile.in prboom-plus-2.5.0.3/src/Makefile.in
--- prboom-plus-2.5.0.3/src/Makefile.in	2009-06-30 02:05:56.000000000 -0700
+++ prboom-plus-2.5.0.3/src/Makefile.in	2009-06-30 02:05:56.000000000 -0700
@@ -268,5 +268,5 @@
 top_srcdir = @top_srcdir@
 SUBDIRS = SDL POSIX MAC PCSOUND TEXTSCREEN
-gamesdir = $(prefix)/games
+gamesdir = $(prefix)/bin
 prboom_plus_game_server_SOURCES = d_server.c protocol.h
 prboom_plus_game_server_LDADD = POSIX/libposixdoom.a SDL/i_network.o @NET_LIBS@ @SDL_LIBS@
