diff -r -U2 gnome-vfs-2.12.0/configure.in gnome-vfs-2.12.0/configure.in
--- gnome-vfs-2.12.0/configure.in	2005-09-05 01:24:22.000000000 -0700
+++ gnome-vfs-2.12.0/configure.in	2005-09-05 01:24:22.000000000 -0700
@@ -518,4 +518,7 @@
 				[AC_DEFINE(HAVE_SAMBA_FLAGS,, [Defined if flags availible in SMBCCTXT])],,
 				[#include <libsmbclient.h>])
+		AC_CHECK_MEMBER(SMBCCTX.close,
+				[AC_DEFINE(HAVE_SAMBA_OLD_CLOSE, , [Defined if old close is available in SMBCCTXT])],,
+				[#include <libsmbclient.h>])
 		CPPFLAGS="$CPPFLAGS_save"
 	else
diff -r -U2 gnome-vfs-2.12.0/modules/smb-method.c gnome-vfs-2.12.0/modules/smb-method.c
--- gnome-vfs-2.12.0/modules/smb-method.c	2005-06-08 00:19:54.000000000 -0700
+++ gnome-vfs-2.12.0/modules/smb-method.c	2005-06-08 00:19:54.000000000 -0700
@@ -1487,5 +1487,9 @@
 		/* Important: perform_authentication leaves and re-enters the lock! */
 		while (perform_authentication (&actx) > 0) {
+#ifdef HAVE_SAMBA_OLD_CLOSE
 			r = smb_context->close (smb_context, handle->file);
+#else
+			r = smb_context->close_fn (smb_context, handle->file);
+#endif
 			actx.res = (r >= 0) ? GNOME_VFS_OK : gnome_vfs_result_from_errno ();
 		}
