diff -r -U2 gkrellm-2.3.2/src/client.c gkrellm-2.3.2/src/client.c
--- gkrellm-2.3.2/src/client.c	2008-09-25 15:11:21.000000000 -0700
+++ gkrellm-2.3.2/src/client.c	2008-09-25 15:11:21.000000000 -0700
@@ -1621,5 +1621,5 @@
 
 static gint
-getline(gint fd, gchar *buf, gint len)
+getline_(gint fd, gchar *buf, gint len)
 	{
 	fd_set			read_fds;
@@ -1715,5 +1715,5 @@
 	while (1)
 		{
-		rs=getline(fd, buf, sizeof(buf));
+		rs=getline_(fd, buf, sizeof(buf));
         if(rs<0)
             return FALSE;
@@ -1733,5 +1733,5 @@
 	while (1)
 		{
-		rs=getline(fd, buf, sizeof(buf));
+		rs=getline_(fd, buf, sizeof(buf));
         if(rs<0)
             return FALSE;
