diff -r -U2 plasticfs-1.8/lib/execve.cc plasticfs-1.8/lib/execve.cc
--- plasticfs-1.8/lib/execve.cc	2004-01-15 17:29:16.000000000 -0700
+++ plasticfs-1.8/lib/execve.cc	2004-01-15 17:29:16.000000000 -0700
@@ -68,5 +68,5 @@
 {
     size_t argv_max = 1024;
-    const char **argv = new (const char *)[argv_max];
+    const char **argv = new const char * [argv_max];
     size_t argc = 0;
     argv[argc++] = arg0;
@@ -86,5 +86,5 @@
 	{
 	    size_t new_argv_max = argv_max * 2;
-	    const char **new_argv = new (const char *)[new_argv_max];
+	    const char **new_argv = new const char * [new_argv_max];
 	    if (!new_argv)
 	    {
diff -r -U2 plasticfs-1.8/lib/itab.cc plasticfs-1.8/lib/itab.cc
--- plasticfs-1.8/lib/itab.cc	2004-01-15 17:29:16.000000000 -0700
+++ plasticfs-1.8/lib/itab.cc	2004-01-15 17:29:16.000000000 -0700
@@ -59,5 +59,5 @@
     hash_cutover_split_mask = (hash_cutover * 2) - 1;
     hash_load = 0;
-    hash_table = new (row_ty *) [hash_modulus];
+    hash_table = new row_ty * [hash_modulus];
     for (hash_ty j = 0; j < hash_modulus; ++j)
 	hash_table[j] = 0;
@@ -78,5 +78,5 @@
     //
     hash_ty new_hash_modulus = hash_modulus + 1;
-    row_ty **new_hash_table = new (row_ty *) [new_hash_modulus];
+    row_ty **new_hash_table = new row_ty * [new_hash_modulus];
     memcpy(new_hash_table, hash_table, hash_modulus * sizeof(row_ty *));
     delete [] hash_table;
diff -r -U2 plasticfs-1.8/lib/plasticfs/execvp.cc plasticfs-1.8/lib/plasticfs/execvp.cc
--- plasticfs-1.8/lib/plasticfs/execvp.cc	2004-01-15 17:29:16.000000000 -0700
+++ plasticfs-1.8/lib/plasticfs/execvp.cc	2004-01-15 17:29:16.000000000 -0700
@@ -49,5 +49,5 @@
     // Construct an argument list for the shell.
     //
-    char **new_argv = new (char *)[argc + 2];
+    char **new_argv = new char * [argc + 2];
     new_argv[0] = _PATH_BSHELL;
     new_argv[1] = (char *)file;
diff -r -U2 plasticfs-1.8/lib/rcstring.cc plasticfs-1.8/lib/rcstring.cc
--- plasticfs-1.8/lib/rcstring.cc	2004-01-15 17:29:16.000000000 -0700
+++ plasticfs-1.8/lib/rcstring.cc	2004-01-15 17:29:16.000000000 -0700
@@ -68,5 +68,5 @@
     hash_mask = hash_modulus - 1;
     hash_load = 0;
-    hash_table = new (string_ty *)[hash_modulus];
+    hash_table = new string_ty * [hash_modulus];
     for (rcstring::hash_ty j = 0; j < hash_modulus; ++j)
 	hash_table[j] = 0;
@@ -90,5 +90,5 @@
     size_t new_hash_modulus = hash_modulus * 2;
     size_t new_hash_mask = new_hash_modulus - 1;
-    string_ty **new_hash_table = new (string_ty *)[new_hash_modulus];
+    string_ty **new_hash_table = new string_ty * [new_hash_modulus];
 
     for (size_t idx = 0; idx < hash_modulus; ++idx)
diff -r -U2 plasticfs-1.8/lib/symtab.cc plasticfs-1.8/lib/symtab.cc
--- plasticfs-1.8/lib/symtab.cc	2004-01-15 17:29:16.000000000 -0700
+++ plasticfs-1.8/lib/symtab.cc	2004-01-15 17:29:16.000000000 -0700
@@ -59,5 +59,5 @@
     hash_cutover_split_mask = (hash_cutover * 2) - 1;
     hash_load = 0;
-    hash_table = new (row_ty *) [hash_modulus];
+    hash_table = new row_ty * [hash_modulus];
     for (hash_ty j = 0; j < hash_modulus; ++j)
 	hash_table[j] = 0;
@@ -78,5 +78,5 @@
     //
     hash_ty new_hash_modulus = hash_modulus + 1;
-    row_ty **new_hash_table = new (row_ty *) [new_hash_modulus];
+    row_ty **new_hash_table = new row_ty * [new_hash_modulus];
     memcpy(new_hash_table, hash_table, hash_modulus * sizeof(row_ty *));
     delete [] hash_table;
