diff -ruN build_mod.old/firegl_public.c build_mod/firegl_public.c
--- build_mod.old/firegl_public.c       2003-12-26 16:12:08.643666840 +1030
+++ build_mod/firegl_public.c   2003-12-26 16:12:59.454942352 +1030
@@ -131,6 +131,9 @@
 #endif
 
 // ============================================================
+#ifndef VMALLOC_VMADDR
+#define VMALLOC_VMADDR(x) ((unsigned long)(x))
+#endif
 #ifndef TRUE
 #define TRUE 1
 #endif

