diff -r -U2 raidtools-1.00.3/detect_multipath.c raidtools-1.00.3/detect_multipath.c
--- raidtools-1.00.3/detect_multipath.c	2003-01-15 01:58:25.000000000 -0700
+++ raidtools-1.00.3/detect_multipath.c	2003-01-15 01:58:25.000000000 -0700
@@ -35,4 +35,6 @@
 #include <string.h>
 #include <fcntl.h>
+typedef unsigned short  sa_family_t;
+#include <linux/netlink.h>
 #include <linux/fs.h>
 
diff -r -U2 raidtools-1.00.3/lsraid.c raidtools-1.00.3/lsraid.c
--- raidtools-1.00.3/lsraid.c	2003-01-15 01:58:25.000000000 -0700
+++ raidtools-1.00.3/lsraid.c	2003-01-15 01:58:25.000000000 -0700
@@ -43,4 +43,6 @@
 #include <linux/types.h>
 #include <linux/major.h>
+typedef unsigned short  sa_family_t;
+#include <linux/netlink.h>
 #include <linux/fs.h>
 #include <linux/raid/md_u.h>
