diff -r -U2 kde-baseapps-4.11.0/CMakeLists.txt kde-baseapps-4.11.0/CMakeLists.txt
--- kde-baseapps-4.11.0/CMakeLists.txt	2013-07-09 15:48:10.000000000 -0700
+++ kde-baseapps-4.11.0/CMakeLists.txt	2013-07-09 15:48:10.000000000 -0700
@@ -14,5 +14,5 @@
 set_package_properties(Strigi PROPERTIES DESCRIPTION "Desktop indexing and search support"
                        URL "http://strigi.sourceforge.net"
-                       TYPE REQUIRED
+                       TYPE OPTIONAL
                       )
 
