diff -r -U2 bash_completion/bash_completion bash_completion/bash_completion
--- bash_completion/bash_completion	2006-03-01 09:20:18.000000000 -0700
+++ bash_completion/bash_completion	2006-03-01 09:20:18.000000000 -0700
@@ -3078,9 +3078,9 @@
 	return 0
 }
-if shopt -q cdable_vars; then
-    complete -v -F _cd $nospace $filenames cd
-else
-    complete -F _cd $nospace $filenames cd
-fi
+#if shopt -q cdable_vars; then
+#    complete -v -F _cd $nospace $filenames cd
+#else
+#    complete -F _cd $nospace $filenames cd
+#fi
 
 # A meta-command completion function for commands like sudo(8), which need to
