diff -r -U2 SDL_ttf-2.0.6/SDL_ttf.c SDL_ttf-2.0.6/SDL_ttf.c
--- SDL_ttf-2.0.6/SDL_ttf.c	2003-02-21 10:38:15.000000000 -0700
+++ SDL_ttf-2.0.6/SDL_ttf.c	2003-02-21 10:38:15.000000000 -0700
@@ -40,8 +40,9 @@
 #endif
 
-#include <freetype/freetype.h>
-#include <freetype/ftoutln.h>
-#include <freetype/ttnameid.h>
-#include <freetype/internal/ftobjs.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
+#include FT_INTERNAL_OBJECTS_H
 
 #include "SDL.h"
@@ -173,5 +174,5 @@
 	  const char*  err_msg;
 	} ft_errors[] = {
-#include <freetype/fterrors.h>
+#include FT_ERRORS_H
 	};
 	int i;
