diff -uN dvd+rw-tools-7.1/transport.hxx dvd+rw-tools-7.1.new/transport.hxx
--- dvd+rw-tools-7.1/transport.hxx	2008-03-01 05:34:43.000000000 -0500
+++ dvd+rw-tools-7.1.new/transport.hxx	2008-04-06 12:37:52.000000000 -0400
@@ -142,6 +142,7 @@
 #include <sys/ioctl.h>
 #include <linux/cdrom.h>
 #include <mntent.h>
+#include <limits.h>
 #include <sys/wait.h>
 #include <sys/utsname.h>
 #include <scsi/sg.h>
